diff --git a/libsn/sn-xutils.c b/libsn/sn-xutils.c
index 220f50e..1278b9f 100644
--- a/libsn/sn-xutils.c
+++ b/libsn/sn-xutils.c
@@ -30,6 +30,25 @@
 #include <xcb/xcb.h>
 #include <xcb/xcb_atom.h>
 
+static xcb_atom_t xcb_atom_get(xcb_connection_t *connection, const char *name)
+{
+    xcb_intern_atom_cookie_t cookie;
+    xcb_intern_atom_reply_t *reply;
+    xcb_atom_t atom;
+
+    if (name == NULL)
+        return XCB_NONE;
+
+    cookie = xcb_intern_atom (connection, 0, strlen(name), name);
+    reply = xcb_intern_atom_reply (connection, cookie, NULL);
+    if (!reply)
+        return XCB_NONE;
+
+    atom = reply->atom;
+    free(reply);
+    return atom;
+}
+
 Atom
 sn_internal_atom_get (SnDisplay  *display,
                       const char *atom_name)
diff --git a/test/test-launcher-xcb.c b/test/test-launcher-xcb.c
index ed06ead..a94c2d3 100644
--- a/test/test-launcher-xcb.c
+++ b/test/test-launcher-xcb.c
@@ -64,7 +64,7 @@ slowly_obtain_timestamp (SnDisplay *display)
     name = "Fake Window";
 
     xcb_change_property (xconnection, XCB_PROP_MODE_REPLACE,
-                         xwindow, WM_NAME, STRING, 8,
+                         xwindow, XCB_ATOM_WM_NAME, XCB_ATOM_STRING, 8,
                          strlen (name), name);
   }
 
