diff --git a/libexslt/Makefile.am b/libexslt/Makefile.am
index 0e92e51..6b1f8f5 100644
--- a/libexslt/Makefile.am
+++ b/libexslt/Makefile.am
@@ -10,9 +10,11 @@ exsltincdir = $(includedir)/libexslt
 
 exsltinc_HEADERS =                      \
 	exslt.h				\
-	exsltconfig.h			\
 	exsltexports.h
 
+nodist_exsltinc_HEADERS =		\
+	exsltconfig.h
+
 libexslt_la_SOURCES =                   \
 	exslt.c				\
 	common.c			\
diff --git a/libexslt/exsltconfig.h b/libexslt/exsltconfig.h
deleted file mode 100644
index 448fc9a..0000000
--- a/libexslt/exsltconfig.h
+++ /dev/null
@@ -1,73 +0,0 @@
-/*
- * exsltconfig.h: compile-time version informations for the EXSLT library
- *
- * See Copyright for the status of this software.
- *
- * daniel@veillard.com
- */
-
-#ifndef __XML_EXSLTCONFIG_H__
-#define __XML_EXSLTCONFIG_H__
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-/**
- * LIBEXSLT_DOTTED_VERSION:
- *
- * the version string like "1.2.3"
- */
-#define LIBEXSLT_DOTTED_VERSION "1.1.28"
-
-/**
- * LIBEXSLT_VERSION:
- *
- * the version number: 1.2.3 value is 10203
- */
-#define LIBEXSLT_VERSION 817
-
-/**
- * LIBEXSLT_VERSION_STRING:
- *
- * the version number string, 1.2.3 value is "10203"
- */
-#define LIBEXSLT_VERSION_STRING "817"
-
-/**
- * LIBEXSLT_VERSION_EXTRA:
- *
- * extra version information, used to show a CVS compilation
- */
-#define	LIBEXSLT_VERSION_EXTRA ""
-
-/**
- * WITH_CRYPTO:
- *
- * Whether crypto support is configured into exslt
- */
-#if 1
-#define EXSLT_CRYPTO_ENABLED
-#endif
-
-/**
- * ATTRIBUTE_UNUSED:
- *
- * This macro is used to flag unused function parameters to GCC
- */
-#ifdef __GNUC__
-#ifdef HAVE_ANSIDECL_H
-#include <ansidecl.h>
-#endif
-#ifndef ATTRIBUTE_UNUSED
-#define ATTRIBUTE_UNUSED __attribute__((unused))
-#endif
-#else
-#define ATTRIBUTE_UNUSED
-#endif
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif /* __XML_EXSLTCONFIG_H__ */
