diff --git a/Makefile.am b/Makefile.am
index 7517cfa..1ff43c8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,8 +15,7 @@ SUBDIRS =								\
 	icons								\
 	pixmaps								\
 	po								\
-	$(PYTHON_SUBDIR)						\
-	tests
+	$(PYTHON_SUBDIR)
 
 AUTOMAKE_OPTIONS =							\
 	1.8								\
diff --git a/Makefile.in b/Makefile.in
index 0c763c6..74b9f9a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -76,7 +76,7 @@ ETAGS = etags
 CTAGS = ctags
 DIST_SUBDIRS = exo exo-csource exo-support exo-desktop-item-edit \
 	exo-gio-module exo-helper exo-open docs icons pixmaps po \
-	python tests
+	python
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -335,8 +335,7 @@ SUBDIRS = \
 	icons								\
 	pixmaps								\
 	po								\
-	$(PYTHON_SUBDIR)						\
-	tests
+	$(PYTHON_SUBDIR)
 
 AUTOMAKE_OPTIONS = \
 	1.8								\
diff --git a/configure b/configure
index d1f8320..d80edc2 100755
--- a/configure
+++ b/configure
@@ -12913,7 +12913,13 @@ fi
 rm -f conftest.mmap conftest.txt
 
 
-if test "$cross_compiling" = yes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether strftime supports %E and %O modifiers." >&5
+$as_echo_n "checking whether strftime supports %E and %O modifiers.... " >&6; }
+if test "${ac_cv_strftime_extension+set}" = set; then :
+  $as_echo_n "(cached) " >&6
+else
+
+		if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
 as_fn_error $? "cannot run test program while cross compiling
@@ -12922,31 +12928,36 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-  #include <string.h>
-  #include <time.h>
-  int
-  main (int argc, char **argv)
-  {
-    struct tm tm;
-    char      buffer[16];
-    tm.tm_year = 81;
-    if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
-      return 0;
-    return 1;
-  }
-
+		  #include <string.h>
+		  #include <time.h>
+		  int
+		  main (int argc, char **argv)
+		  {
+		    struct tm tm;
+		    char      buffer[16];
+		    tm.tm_year = 81;
+		    if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
+		      return 0;
+		    return 1;
+		  }
 _ACEOF
 if ac_fn_c_try_run "$LINENO"; then :
-
-
-$as_echo "#define HAVE_STRFTIME_EXTENSION 1" >>confdefs.h
-
-
+  ac_cv_strftime_extension=yes
+else
+  ac_cv_strftime_extension=no
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
 fi
 
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_strftime_extension" >&5
+$as_echo "$ac_cv_strftime_extension" >&6; }
+if test "x$ac_cv_strftime_extension" = "xyes"; then
+
+$as_echo "#define HAVE_STRFTIME_EXTENSION 1" >>confdefs.h
+
+fi
 
 
     GETTEXT_PACKAGE=exo-1
diff --git a/configure.in b/configure.in
index 97034fa..5e7241d 100644
--- a/configure.in
+++ b/configure.in
@@ -124,22 +124,27 @@ AC_FUNC_MMAP()
 dnl ***************************************
 dnl *** Check for strftime() extensions ***
 dnl ***************************************
-AC_TRY_RUN([
-  #include <string.h>
-  #include <time.h>
-  int
-  main (int argc, char **argv)
-  {
-    struct tm tm;
-    char      buffer[16];
-    tm.tm_year = 81;
-    if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
-      return 0;
-    return 1;
-  }
-], [
+AC_CACHE_CHECK([whether strftime supports %E and %O modifiers.],
+	ac_cv_strftime_extension,[
+		AC_TRY_RUN([
+		  #include <string.h>
+		  #include <time.h>
+		  int
+		  main (int argc, char **argv)
+		  {
+		    struct tm tm;
+		    char      buffer[16];
+		    tm.tm_year = 81;
+		    if (strftime (buffer, 16, "%EY", &tm) == 4 && strcmp (buffer, "1981") == 0)
+		      return 0;
+		    return 1;
+		  }],
+		  [ac_cv_strftime_extension=yes],
+		  [ac_cv_strftime_extension=no],
+		  [])])
+if test "x$ac_cv_strftime_extension" = "xyes"; then
   AC_DEFINE([HAVE_STRFTIME_EXTENSION], 1, [Define if strftime supports %E and %O modifiers.])
-])
+fi
 
 dnl ******************************
 dnl *** Check for i18n support ***
