diff --git a/Makefile.am b/Makefile.am
index dd76c7e..6f53c37 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC=dbus-glib-1.pc
 
-SUBDIRS=dbus tools test doc
+SUBDIRS=dbus
 DIST_SUBDIRS=dbus tools test doc m4
 
 pkgconfigdir = $(libdir)/pkgconfig
diff --git a/Makefile.in b/Makefile.in
index 2180163..c81ab19 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -405,7 +405,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC = dbus-glib-1.pc
-SUBDIRS = dbus tools test doc
+SUBDIRS = dbus
 DIST_SUBDIRS = dbus tools test doc m4
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index 6433224..ac8b82f 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)						\
diff --git a/dbus/Makefile.in b/dbus/Makefile.in
index bbf1306..ca4ec99 100644
--- a/dbus/Makefile.in
+++ b/dbus/Makefile.in
@@ -611,7 +611,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . examples
+SUBDIRS = .
 AM_CPPFLAGS = \
 	-I$(top_srcdir)						\
 	-I$(top_builddir)					\
