diff --git a/extra/config/confdata.c b/extra/config/confdata.c
index 2b848d5..a40d5fa 100644
--- a/extra/config/confdata.c
+++ b/extra/config/confdata.c
@@ -394,7 +394,7 @@ int conf_write(const char *name)
 	struct symbol *sym;
 	struct menu *menu;
 	const char *basename;
-	char dirname[128], tmpname[128], newname[128];
+	char dirname[PATH_MAX+1], tmpname[PATH_MAX+1], newname[PATH_MAX+1];
 	int type, l;
 	const char *str;
 	time_t now;
