From 661f1d4c3a390c42027a91a8731cf31edefcfb71 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Thu, 12 Jan 2012 08:11:33 +0100
Subject: [PATCH] Check for SDL via pkg-config.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 configure.ac |   24 +-----------------------
 1 files changed, 1 insertions(+), 23 deletions(-)

diff --git a/configure.ac b/configure.ac
index d74b3c1..c94f7a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -585,30 +585,8 @@ AM_CONDITIONAL(WITH_TIGHTVNC_FILETRANSFER, test "$with_tightvnc_filetransfer" =
 AM_CONDITIONAL(HAVE_LIBZ, test ! -z "$HAVE_ZLIB_H")
 AM_CONDITIONAL(HAVE_LIBJPEG, test ! -z "$HAVE_JPEGLIB_H")
 
-
-SDLCONFIG="sdl-config"
-AC_ARG_WITH(sdl-config,
-[[  --with-sdl-config=FILE
-    Use the given path to sdl-config when determining
-    SDL configuration; defaults to "sdl-config"]],
-[
-    if test "$withval" != "yes" -a "$withval" != ""; then
-        SDLCONFIG=$withval
-    fi
-])
-
-if test -z "$with_sdl"; then
-	if $SDLCONFIG --version >/dev/null 2>&1; then
-		with_sdl=yes
-		SDL_CFLAGS=`$SDLCONFIG --cflags`
-		SDL_LIBS=`$SDLCONFIG --libs`
-	else
-		with_sdl=no
-	fi
-fi
+PKG_CHECK_MODULES(SDL, sdl, [with_sdl=yes], [with_sdl=no])
 AM_CONDITIONAL(HAVE_LIBSDL, test "x$with_sdl" = "xyes")
-AC_SUBST(SDL_CFLAGS)
-AC_SUBST(SDL_LIBS)
 
 MINGW=`echo $host_os | grep mingw32  2>/dev/null`
 AM_CONDITIONAL(MINGW, test ! -z "$MINGW" )
-- 
1.7.8.3

