diff --git a/Makefile b/Makefile
index 77140eb..f9ecbf8 100644
--- a/Makefile
+++ b/Makefile
@@ -453,10 +453,6 @@ $(obj)mmc_spl/u-boot-mmc-spl.bin:	mmc_spl
 $(obj)spl/u-boot-spl.bin:		depend
 		$(MAKE) -C spl all
 
-$(TIMESTAMP_FILE):
-		@LC_ALL=C date +'#define U_BOOT_DATE "%b %d %C%y"' > $@
-		@LC_ALL=C date +'#define U_BOOT_TIME "%T"' >> $@
-
 updater:
 		$(MAKE) -C tools/updater all
 
@@ -547,7 +543,7 @@ $(obj)$(CPUDIR)/$(SOC)/asm-offsets.s:	$(obj)include/autoconf.mk.dep
 else	# !config.mk
 all $(obj)u-boot.hex $(obj)u-boot.srec $(obj)u-boot.bin \
 $(obj)u-boot.img $(obj)u-boot.dis $(obj)u-boot \
-$(filter-out tools,$(SUBDIRS)) $(TIMESTAMP_FILE) \
+$(filter-out tools,$(SUBDIRS)) \
 updater depend dep tags ctags etags cscope $(obj)System.map:
 	@echo "System not configured - see README" >&2
 	@ exit 1
@@ -570,6 +566,10 @@ $(VERSION_FILE):
 		 '$(shell $(LD) -v | head -n 1)' )>>  $@.tmp
 		@cmp -s $@ $@.tmp && rm -f $@.tmp || mv -f $@.tmp $@
 
+$(TIMESTAMP_FILE):
+		@LC_ALL=C date +'#define U_BOOT_DATE "%b %d %C%y"' > $@
+		@LC_ALL=C date +'#define U_BOOT_TIME "%T"' >> $@
+
 easylogo env gdb:
 	$(MAKE) -C tools/$@ all MTD_VERSION=${MTD_VERSION}
 gdbtools: gdb
diff --git a/tools/Makefile b/tools/Makefile
index fc741d3..9b3583d 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -158,6 +158,9 @@ HOSTCPPFLAGS =	-idirafter $(SRCTREE)/include \
 		-DUSE_HOSTCC \
 		-D__KERNEL_STRICT_NAMES
 
+ifeq ($(wildcard $(obj)include/config.h),$(obj)include/config.h)
+  HOSTCPPFLAGS += -DHAVE_CONFIG_H=1
+endif
 
 all:	$(obj).depend $(BINS) $(LOGO-y) subdirs
 
diff --git a/tools/ublimage.h b/tools/ublimage.h
index c926689..433cd94 100644
--- a/tools/ublimage.h
+++ b/tools/ublimage.h
@@ -28,7 +28,9 @@
 #ifndef _UBLIMAGE_H_
 #define _UBLIMAGE_H_
 
-#include <config.h>
+#ifdef HAVE_CONFIG_H
+#  include <config.h>
+#endif
 
 #if !defined(CONFIG_SYS_UBL_BLOCK)
 #define CONFIG_SYS_UBL_BLOCK 512
