commit d31c5375cb6411c4494fa5d8fab5b4e4bae3c05d
Author: Thierry Reding <thierry.reding@avionic-design.de>
Date:   Thu Jul 2 12:08:53 2009 +0200

    Fix automake warning.

diff --git a/sxpm/Makefile.am b/sxpm/Makefile.am
index 7780bd8..65388f8 100644
--- a/sxpm/Makefile.am
+++ b/sxpm/Makefile.am
@@ -18,8 +18,6 @@ appman_DATA = $(appman_PRE:man=@APP_MAN_SUFFIX@)
 
 CLEANFILES = $(appman_DATA)
 
-SED = sed
-
 # Strings to replace in man pages
 XORGRELSTRING = @PACKAGE_STRING@
   XORGMANNAME = X Version 11
