diff --git a/Makefile.in b/Makefile.in
index 45f2972..ef860ab 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -9,16 +9,13 @@ INSTALL = @INSTALL@
 
 % : %.sh
 
-@RESIZER_CMT@RESIZE_DIR= resize
-@DEBUGFS_CMT@DEBUGFS_DIR= debugfs
-@UUID_CMT@UUID_LIB_SUBDIR= lib/uuid
-@BLKID_CMT@BLKID_LIB_SUBDIR= lib/blkid
+UUID_LIB_SUBDIR= lib/uuid
 
-LIB_SUBDIRS=lib/et lib/ss lib/e2p $(UUID_LIB_SUBDIR) lib/ext2fs $(BLKID_LIB_SUBDIR) intl
-PROG_SUBDIRS=e2fsck $(DEBUGFS_DIR) misc $(RESIZE_DIR) tests/progs po
-SUBDIRS=util $(LIB_SUBDIRS) $(PROG_SUBDIRS) tests
+LIB_SUBDIRS=$(UUID_LIB_SUBDIR)
+PROG_SUBDIRS=
+SUBDIRS=$(LIB_SUBDIRS)
 
-SUBS= lib/ext2fs/ext2_types.h lib/blkid/blkid_types.h lib/uuid/uuid_types.h
+SUBS= lib/uuid/uuid_types.h
 
 TAR=tar
 
diff --git a/lib/uuid/Makefile.in b/lib/uuid/Makefile.in
index e8c5a21..1fe7465 100644
--- a/lib/uuid/Makefile.in
+++ b/lib/uuid/Makefile.in
@@ -129,7 +129,7 @@ installdirs::
 	$(E) "	MKINSTALLDIRS $(libdir) $(includedir)/uuid $(man3dir)"
 	$(Q) $(MKINSTALLDIRS) $(DESTDIR)$(libdir)  \
 		$(DESTDIR)$(includedir)/uuid $(DESTDIR)$(man3dir) \
-		$(DESTDIR)$(libdir)/pkgconfig
+		$(DESTDIR)$(prefix)/lib/pkgconfig
 
 install:: all installdirs 
 	$(E) "	INSTALL_DATA $(libdir)/libuuid.a"
@@ -149,12 +149,12 @@ install:: all installdirs
 	$(Q) $(LN) -f $(DESTDIR)$(man3dir)/uuid_generate.3 $(DESTDIR)$(man3dir)/uuid_generate_random.3
 	$(E) "	LINK $(man3dir)/uuid_generate_time.3"
 	$(Q) $(LN) -f $(DESTDIR)$(man3dir)/uuid_generate.3 $(DESTDIR)$(man3dir)/uuid_generate_time.3
-	$(E) "	INSTALL_DATA $(libdir)/pkgconfig/uuid.pc"
-	$(Q) $(INSTALL_DATA) uuid.pc $(DESTDIR)$(libdir)/pkgconfig/uuid.pc
+	$(E) "	INSTALL_DATA $(prefix)/lib/pkgconfig/uuid.pc"
+	$(Q) $(INSTALL_DATA) uuid.pc $(DESTDIR)$(prefix)/lib/pkgconfig/uuid.pc
 
 uninstall::
 	$(RM) -f $(DESTDIR)$(libdir)/libuuid.a \
-		$(DESTDIR)$(libdir)/pkgconfig/uuid.pc
+		$(DESTDIR)$(prefix)/lib/pkgconfig/uuid.pc
 	for i in $(SMANPAGES); do \
 		$(RM) -f $(DESTDIR)$(man3dir)/$$i; \
 	done
