From 62281fc400614a4522d1029afe6d0deeab449290 Mon Sep 17 00:00:00 2001
From: Julian Scheel <julian.scheel@avionic-design.de>
Date: Mon, 9 Jan 2012 13:58:37 +0100
Subject: [PATCH] gnutls: Fix potential NULL-pointer dereference.

Signed-off-by: Julian Scheel <julian.scheel@avionic-design.de>
---
 tls/gnutls/gtlsconnection-gnutls.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/tls/gnutls/gtlsconnection-gnutls.c b/tls/gnutls/gtlsconnection-gnutls.c
index 4bed055..3da7d7a 100644
--- a/tls/gnutls/gtlsconnection-gnutls.c
+++ b/tls/gnutls/gtlsconnection-gnutls.c
@@ -805,6 +805,8 @@ handshake_internal (GTlsConnectionGnutls  *gnutls,
   GTlsCertificateFlags peer_certificate_errors = 0;
   int ret;
 
+  g_return_val_if_fail (G_IS_TLS_CONNECTION_GNUTLS (gnutls), FALSE);
+
   if (G_IS_TLS_SERVER_CONNECTION_GNUTLS (gnutls) &&
       gnutls->priv->ever_handshaked && !gnutls->priv->handshaking &&
       !gnutls->priv->need_handshake)
-- 
1.7.9

