--- a/configure.ac	2007-06-27 13:54:08.000000000 +0200
+++ b/configure.ac	2007-06-27 13:54:53.000000000 +0200
@@ -50,7 +50,7 @@
 
 # Find the xcb-proto protocol descriptions
 AC_MSG_CHECKING(XCBPROTO_XCBINCLUDEDIR)
-XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
+XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --define-variable=prefix=${DESTDIR}${prefix} --variable=xcbincludedir xcb-proto`
 AC_MSG_RESULT($XCBPROTO_XCBINCLUDEDIR)
 AC_SUBST(XCBPROTO_XCBINCLUDEDIR)
 
--- a/configure	2007-06-27 13:55:58.000000000 +0200
+++ b/configure	2007-06-27 13:56:25.000000000 +0200
@@ -21211,7 +21211,7 @@
 # Find the xcb-proto protocol descriptions
 { echo "$as_me:$LINENO: checking XCBPROTO_XCBINCLUDEDIR" >&5
 echo $ECHO_N "checking XCBPROTO_XCBINCLUDEDIR... $ECHO_C" >&6; }
-XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
+XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --define-variable=prefix=${DESTDIR}${prefix} --variable=xcbincludedir xcb-proto`
 { echo "$as_me:$LINENO: result: $XCBPROTO_XCBINCLUDEDIR" >&5
 echo "${ECHO_T}$XCBPROTO_XCBINCLUDEDIR" >&6; }
 
