diff --git a/configure.in b/configure.in
index 9b2368d..1c1ba78 100644
--- a/configure.in
+++ b/configure.in
@@ -1490,6 +1490,18 @@ AS_AC_EXPAND(SYSCONFDIR, $sysconfdir)
 
 CFLAGS="$CFLAGS -Werror-implicit-function-declaration"
 
+# by Marcelo Magallon <mmagallo@efis.ucr.ac.cr>
+# Turn around -rpath problem with libtool 1.0c
+# This define should be improbable enough to not conflict with anything
+case ${host} in
+  *-linux-*)
+    AC_MSG_RESULT([Fixing libtool for -rpath problems.])
+    sed < libtool > libtool-2 \
+    's/^hardcode_libdir_flag_spec.*$'/'hardcode_libdir_flag_spec=" -D__LIBTOOL_IS_A_FOOL__ "/'
+    mv libtool-2 libtool
+    chmod 755 libtool
+  ;;
+esac
 
 AC_OUTPUT([
 
