commit 63144636eb3d89960f99bc0c5d81535c39b15320
Author: Thierry Reding <thierry@avionic-tr-basi.adnet.avionic-design.de>
Date:   Thu May 29 12:57:09 2008 +0200

    Add rpl_malloc() implementation.

diff --git a/src/Makefile.am b/src/Makefile.am
index 1613941..495dd64 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -18,7 +18,7 @@ include_HEADERS  = tslib.h
 lib_LTLIBRARIES  = libts.la
 libts_la_SOURCES = ts_attach.c ts_close.c ts_config.c ts_error.c \
 		   ts_fd.c ts_load_module.c ts_open.c ts_parse_vars.c \
-		   ts_read.c ts_read_raw.c
+		   ts_read.c ts_read_raw.c rpl_malloc.c
 libts_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 		   -release $(LT_RELEASE) -export-dynamic
 libts_la_LIBADD  = -ldl
diff --git a/src/rpl_malloc.c b/src/rpl_malloc.c
new file mode 100644
index 0000000..c8a12ad
--- /dev/null
+++ b/src/rpl_malloc.c
@@ -0,0 +1,14 @@
+#include "config.h"
+
+#include <stdlib.h>
+
+#if malloc == rpl_malloc
+void *rpl_malloc(size_t size)
+{
+	if (size == 0)
+		size = 1;
+
+	return malloc(size);
+}
+#endif
+
