diff --git a/Makefile b/Makefile
index b5c94ab..094474b 100644
--- a/Makefile
+++ b/Makefile
@@ -90,8 +90,8 @@ include/config.h: include/builddefs
 	fi
 
 install: default $(addsuffix -install,$(SUBDIRS))
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 README $(PKG_DOC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 README $(DESTDIR)$(PKG_DOC_DIR)
 
 install-dev: default $(addsuffix -install-dev,$(SUBDIRS))
 
diff --git a/attr/Makefile b/attr/Makefile
index 1c467e8..6fc245b 100644
--- a/attr/Makefile
+++ b/attr/Makefile
@@ -29,6 +29,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
diff --git a/doc/Makefile b/doc/Makefile
index ab3cdf8..d555950 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -32,12 +32,12 @@ CHANGES.gz:
 
 install: default
 	$(SUBDIRS_MAKERULE)
-	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
-	$(INSTALL) -m 644 PORTING CHANGES.gz $(PKG_DOC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DOC_DIR)
+	$(INSTALL) -m 644 PORTING CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)
 ifeq ($(PKG_DISTRIBUTION), debian)
-	$(INSTALL) -S CHANGES.gz $(PKG_DOC_DIR)/changelog.gz
+	$(INSTALL) -S CHANGES.gz $(DESTDIR)$(PKG_DOC_DIR)/changelog.gz
 else
-	$(INSTALL) -m 644 COPYING $(PKG_DOC_DIR)
+	$(INSTALL) -m 644 COPYING $(DESTDIR)$(PKG_DOC_DIR)
 endif
 
 install-dev install-lib: $(SUBDIRS)
diff --git a/getfattr/Makefile b/getfattr/Makefile
index 91d3df2..e162f4b 100644
--- a/getfattr/Makefile
+++ b/getfattr/Makefile
@@ -30,6 +30,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
diff --git a/include/Makefile b/include/Makefile
index 3068e56..4471bcf 100644
--- a/include/Makefile
+++ b/include/Makefile
@@ -33,7 +33,7 @@ $(INCDIR):
 include $(BUILDRULES)
 
 install-dev: default
-	$(INSTALL) -m 755 -d $(PKG_INC_DIR)
-	$(INSTALL) -m 644 $(INST_HFILES) $(PKG_INC_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_INC_DIR)
+	$(INSTALL) -m 644 $(INST_HFILES) $(DESTDIR)$(PKG_INC_DIR)
 
 install install-lib:
diff --git a/include/builddefs.in b/include/builddefs.in
index 6df0f5a..885a069 100644
--- a/include/builddefs.in
+++ b/include/builddefs.in
@@ -96,7 +96,7 @@ GCFLAGS = $(OPTIMIZER) $(DEBUG) -funsigned-char -fno-strict-aliasing -Wall \
 	  -DPACKAGE=\"$(PKG_NAME)\" -I$(TOPDIR)/include
 
 # Global, Platform, Local CFLAGS
-CFLAGS += $(GCFLAGS) $(PCFLAGS) $(LCFLAGS)
+CFLAGS = @CFLAGS@ $(GCFLAGS) $(PCFLAGS) $(LCFLAGS)
 
 include $(TOPDIR)/include/buildmacros
 
diff --git a/include/buildmacros b/include/buildmacros
index ab89182..05a6dda 100644
--- a/include/buildmacros
+++ b/include/buildmacros
@@ -68,9 +68,9 @@ endif
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(PKG_LIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -m 644 -T so_dot_version $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -T so_dot_current $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR)
 endif
 
 # Libtool thinks the static and shared libs should be in the same dir, so
@@ -82,15 +82,15 @@ endif
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB_DEV = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 $(LIBNAME).lai $(PKG_DEVLIB_DIR)/$(LIBNAME).la ; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -T so_base $(LIBNAME).lai $(PKG_LIB_DIR); \
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DEVLIB_DIR); \
+	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(DESTDIR)$(PKG_DEVLIB_DIR); \
+	../$(INSTALL) -m 644 $(LIBNAME).lai $(DESTDIR)$(PKG_DEVLIB_DIR)/$(LIBNAME).la ; \
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_LIB_DIR); \
+	../$(INSTALL) -T so_base $(LIBNAME).lai $(DESTDIR)$(PKG_LIB_DIR); \
 	if test "x$(PKG_DEVLIB_DIR)" != "x$(PKG_LIB_DIR)" ; then \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).a $(PKG_LIB_DIR)/$(LIBNAME).a; \
-	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).la $(PKG_LIB_DIR)/$(LIBNAME).la; \
-	../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).so $(PKG_DEVLIB_DIR)/$(LIBNAME).so; \
+	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).a $(DESTDIR)$(PKG_LIB_DIR)/$(LIBNAME).a; \
+	../$(INSTALL) -S $(PKG_DEVLIB_DIR)/$(LIBNAME).la $(DESTDIR)$(PKG_LIB_DIR)/$(LIBNAME).la; \
+	../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).so $(DESTDIR)$(PKG_DEVLIB_DIR)/$(LIBNAME).so; \
 	fi
 else
 INSTALL_LTLIB_DEV = $(INSTALL_LTLIB_STATIC)
@@ -98,8 +98,8 @@ endif
 
 INSTALL_LTLIB_STATIC = \
 	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_DEVLIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_DEVLIB_DIR)
+	../$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_DEVLIB_DIR); \
+	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(DESTDIR)$(PKG_DEVLIB_DIR)
 
 INSTALL_MAN = \
 	@for d in $(MAN_PAGES); do \
@@ -117,11 +117,11 @@ INSTALL_MAN = \
 					$(ZIP) -9 -c $$d > $$d.gz; _sfx=.gz; \
 				fi; \
 				u=$$m.$(MAN_SECTION)$$_sfx; \
-				echo $(INSTALL) -m 644 $${d}$$_sfx $${t}$$_sfx;\
-				$(INSTALL) -m 644 $${d}$$_sfx $${t}$$_sfx; \
+				echo $(INSTALL) -m 644 $${d}$$_sfx $(DESTDIR)$${t}$$_sfx;\
+				$(INSTALL) -m 644 $${d}$$_sfx $(DESTDIR)$${t}$$_sfx; \
 			else \
-				echo $(INSTALL) -S $$u $${t}$$_sfx; \
-				$(INSTALL) -S $$u $${t}$$_sfx; \
+				echo $(INSTALL) -S $$u $(DESTDIR)$${t}$$_sfx; \
+				$(INSTALL) -S $$u $(DESTDIR)$${t}$$_sfx; \
 			fi; \
 			first=false; \
 		done; \
@@ -132,8 +132,8 @@ INSTALL_LINGUAS = \
 	@for l in $(LINGUAS) ""; do \
 		if test -f "$$l.mo" ; then \
 			ldir=$(PKG_LOCALE_DIR)/$$l/LC_MESSAGES; \
-			$(INSTALL) -m 755 -d $$ldir; \
-			$(INSTALL) -m 644 $$l.mo $$ldir/$(PKG_NAME).mo; \
+			$(INSTALL) -m 755 -d $(DESTDIR)$$ldir; \
+			$(INSTALL) -m 644 $$l.mo $(DESTDIR)$$ldir/$(PKG_NAME).mo; \
 		fi; \
 	done
 endif
diff --git a/man/man1/Makefile b/man/man1/Makefile
index 04db140..5e6905f 100644
--- a/man/man1/Makefile
+++ b/man/man1/Makefile
@@ -30,6 +30,6 @@ default : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-dev install-lib:
diff --git a/man/man2/Makefile b/man/man2/Makefile
index d77309d..8b252e2 100644
--- a/man/man2/Makefile
+++ b/man/man2/Makefile
@@ -30,6 +30,6 @@ default install : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install-dev : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-lib:
diff --git a/man/man3/Makefile b/man/man3/Makefile
index 4a07f5f..a19142c 100644
--- a/man/man3/Makefile
+++ b/man/man3/Makefile
@@ -30,6 +30,6 @@ default install : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install-dev : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-lib:
diff --git a/man/man5/Makefile b/man/man5/Makefile
index 6b70d3d..6f5387b 100644
--- a/man/man5/Makefile
+++ b/man/man5/Makefile
@@ -30,6 +30,6 @@ default : $(MAN_PAGES)
 include $(BUILDRULES)
 
 install : default
-	$(INSTALL) -m 755 -d $(MAN_DEST)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(MAN_DEST)
 	$(INSTALL_MAN)
 install-dev install-lib:
diff --git a/setfattr/Makefile b/setfattr/Makefile
index d55461b..2f9cf37 100644
--- a/setfattr/Makefile
+++ b/setfattr/Makefile
@@ -30,6 +30,6 @@ default: $(LTCOMMAND)
 include $(BUILDRULES)
 
 install: default
-	$(INSTALL) -m 755 -d $(PKG_BIN_DIR)
-	$(LTINSTALL) -m 755 $(LTCOMMAND) $(PKG_BIN_DIR)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(PKG_BIN_DIR)
+	$(LTINSTALL) -m 755 $(LTCOMMAND) $(DESTDIR)$(PKG_BIN_DIR)
 install-dev install-lib:
