From 72aaed7b0f5f1a108fadf2c0934c1384eb7a9fc6 Mon Sep 17 00:00:00 2001
From: Dirk Leber <dirk.leber@avionic-design.de>
Date: Mon, 26 Jan 2015 09:58:03 +0100
Subject: [PATCH 3/4] upd: Fix to enable smoth initial playback

Signed-off-by: Dirk Leber <dirk.leber@avionic-design.de>
---
 modules/access/udp.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/access/udp.c b/modules/access/udp.c
index 76f68a2..895c934 100644
--- a/modules/access/udp.c
+++ b/modules/access/udp.c
@@ -209,11 +209,15 @@ static int Control( access_t *p_access, int i_query, va_list args )
         case ACCESS_CAN_SEEK:
         case ACCESS_CAN_FASTSEEK:
         case ACCESS_CAN_PAUSE:
-        case ACCESS_CAN_CONTROL_PACE:
             pb_bool = (bool*)va_arg( args, bool* );
             *pb_bool = false;
             break;
 
+        case ACCESS_CAN_CONTROL_PACE:
+            pb_bool = (bool*)va_arg( args, bool* );
+            *pb_bool = true;
+            break;
+
         case ACCESS_GET_PTS_DELAY:
             pi_64 = (int64_t*)va_arg( args, int64_t * );
             *pi_64 = INT64_C(1000)
-- 
2.1.0

