Autoreconfigure.

From: Thierry Reding <thierry.reding@avionic-design.de>


---
 configure |   56 ++++++++++++++++++++++++++++----------------------------
 1 files changed, 28 insertions(+), 28 deletions(-)

diff --git a/configure b/configure
index 9126819..82e8d00 100755
--- a/configure
+++ b/configure
@@ -27550,12 +27550,12 @@ if test -n "$NOTIFY_CFLAGS"; then
     pkg_cv_NOTIFY_CFLAGS="$NOTIFY_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libnotify") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify >= 0.7.0 gtk+-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libnotify >= 0.7.0 gtk+-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_NOTIFY_CFLAGS=`$PKG_CONFIG --cflags "libnotify" 2>/dev/null`
+  pkg_cv_NOTIFY_CFLAGS=`$PKG_CONFIG --cflags "libnotify >= 0.7.0 gtk+-2.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -27566,12 +27566,12 @@ if test -n "$NOTIFY_LIBS"; then
     pkg_cv_NOTIFY_LIBS="$NOTIFY_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libnotify") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify >= 0.7.0 gtk+-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libnotify >= 0.7.0 gtk+-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_NOTIFY_LIBS=`$PKG_CONFIG --libs "libnotify" 2>/dev/null`
+  pkg_cv_NOTIFY_LIBS=`$PKG_CONFIG --libs "libnotify >= 0.7.0 gtk+-2.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -27591,28 +27591,28 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libnotify" 2>&1`
+	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libnotify >= 0.7.0 gtk+-2.0" 2>&1`
         else
-	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors "libnotify" 2>&1`
+	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors "libnotify >= 0.7.0 gtk+-2.0" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$NOTIFY_PKG_ERRORS" >&5
 
 	if test x"$enable_notify" = "xyes"; then :
-  as_fn_error $? "Library libnotify needed for notify was not found" "$LINENO" 5
+  as_fn_error $? "Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" "$LINENO" 5
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify needed for notify was not found" >&5
-$as_echo "$as_me: WARNING: Library libnotify needed for notify was not found" >&2;}
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&5
+$as_echo "$as_me: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&2;}
 
 fi
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
 	if test x"$enable_notify" = "xyes"; then :
-  as_fn_error $? "Library libnotify needed for notify was not found" "$LINENO" 5
+  as_fn_error $? "Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" "$LINENO" 5
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify needed for notify was not found" >&5
-$as_echo "$as_me: WARNING: Library libnotify needed for notify was not found" >&2;}
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&5
+$as_echo "$as_me: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&2;}
 
 fi
 else
@@ -27649,12 +27649,12 @@ if test -n "$NOTIFY_CFLAGS"; then
     pkg_cv_NOTIFY_CFLAGS="$NOTIFY_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libnotify") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify >= 0.7.0 gtk+-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libnotify >= 0.7.0 gtk+-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_NOTIFY_CFLAGS=`$PKG_CONFIG --cflags "libnotify" 2>/dev/null`
+  pkg_cv_NOTIFY_CFLAGS=`$PKG_CONFIG --cflags "libnotify >= 0.7.0 gtk+-2.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -27665,12 +27665,12 @@ if test -n "$NOTIFY_LIBS"; then
     pkg_cv_NOTIFY_LIBS="$NOTIFY_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libnotify") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libnotify >= 0.7.0 gtk+-2.0\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libnotify >= 0.7.0 gtk+-2.0") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_NOTIFY_LIBS=`$PKG_CONFIG --libs "libnotify" 2>/dev/null`
+  pkg_cv_NOTIFY_LIBS=`$PKG_CONFIG --libs "libnotify >= 0.7.0 gtk+-2.0" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -27690,19 +27690,19 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libnotify" 2>&1`
+	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "libnotify >= 0.7.0 gtk+-2.0" 2>&1`
         else
-	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors "libnotify" 2>&1`
+	        NOTIFY_PKG_ERRORS=`$PKG_CONFIG --print-errors "libnotify >= 0.7.0 gtk+-2.0" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$NOTIFY_PKG_ERRORS" >&5
 
 	enable_notify=no
  if test x"$enable_notify" = "xyes"; then :
-  as_fn_error $? "Library libnotify needed for notify was not found" "$LINENO" 5
+  as_fn_error $? "Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" "$LINENO" 5
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify needed for notify was not found" >&5
-$as_echo "$as_me: WARNING: Library libnotify needed for notify was not found" >&2;}
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&5
+$as_echo "$as_me: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&2;}
 
 fi
 elif test $pkg_failed = untried; then
@@ -27710,10 +27710,10 @@ elif test $pkg_failed = untried; then
 $as_echo "no" >&6; }
 	enable_notify=no
  if test x"$enable_notify" = "xyes"; then :
-  as_fn_error $? "Library libnotify needed for notify was not found" "$LINENO" 5
+  as_fn_error $? "Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" "$LINENO" 5
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify needed for notify was not found" >&5
-$as_echo "$as_me: WARNING: Library libnotify needed for notify was not found" >&2;}
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&5
+$as_echo "$as_me: WARNING: Library libnotify >= 0.7.0 gtk+-2.0 needed for notify was not found" >&2;}
 
 fi
 else
