diff --git a/modules/access/v4l2.c b/modules/access/v4l2.c
index d1aafc0..86064dd 100644
--- a/modules/access/v4l2.c
+++ b/modules/access/v4l2.c
@@ -709,6 +709,8 @@ static int DemuxOpen( vlc_object_t *p_this )
  *****************************************************************************/
 static void GetV4L2Params( demux_sys_t *p_sys, vlc_object_t *p_obj )
 {
+    p_sys->psz_device = var_CreateGetNonEmptyString( p_obj, "v4l2-dev" );
+
     p_sys->i_video_pts = -1;
 
     p_sys->i_selected_standard_id =
@@ -748,7 +750,6 @@ static void GetV4L2Params( demux_sys_t *p_sys, vlc_object_t *p_obj )
     }
     free( psz_aspect );
 
-    p_sys->psz_device = NULL;
     p_sys->i_fd = -1;
 
     p_sys->p_es = NULL;
@@ -1007,7 +1008,7 @@ static void ParseMRL( demux_sys_t *p_sys, char *psz_path, vlc_object_t *p_obj )
     /* Main device */
     if( *psz_dup )
         p_sys->psz_device = strdup( psz_dup );
-    else
+    else if( p_sys->psz_device == NULL )
         p_sys->psz_device = strdup( V4L2_DEFAULT );
     free( psz_dup );
 }
