diff --git a/modules/access/v4l2.c b/modules/access/v4l2.c
index e899a04..952191f 100644
--- a/modules/access/v4l2.c
+++ b/modules/access/v4l2.c
@@ -1429,6 +1429,18 @@ static block_t* GrabVideo( vlc_object_t *p_demux, demux_sys_t *p_sys )
                }
         }
 
+        if( buf.flags & V4L2_BUF_FLAG_ERROR ) {
+            msg_Err( p_demux, "Invalid buffer received" );
+
+            if( v4l2_ioctl( p_sys->i_fd, VIDIOC_QBUF, &buf ) < 0 )
+            {
+                msg_Err( p_demux, "Failed to unlock (VIDIOC_QBUF)" );
+                block_Release( p_block );
+            }
+
+            return NULL;
+        }
+
         if( buf.index >= p_sys->i_nbuffers ) {
             msg_Err( p_demux, "Failed capturing new frame as i>=nbuffers" );
             return NULL;
@@ -1469,6 +1481,18 @@ static block_t* GrabVideo( vlc_object_t *p_demux, demux_sys_t *p_sys )
             }
         }
 
+        if( buf.flags & V4L2_BUF_FLAG_ERROR ) {
+            msg_Err( p_demux, "Invalid buffer received" );
+
+            if( v4l2_ioctl( p_sys->i_fd, VIDIOC_QBUF, &buf ) < 0 )
+            {
+                msg_Err( p_demux, "Failed to unlock (VIDIOC_QBUF)" );
+                block_Release( p_block );
+            }
+
+            return NULL;
+        }
+
         /* Find frame? */
         unsigned int i;
         for( i = 0; i < p_sys->i_nbuffers; i++ )
