diff --git a/source/aclocal.m4 b/source/aclocal.m4
index 6274134..f38393f 100644
--- a/source/aclocal.m4
+++ b/source/aclocal.m4
@@ -152,7 +152,7 @@ AC_DEFUN(AC_CHECK_64BIT_LIBS,
     DEFAULT_64BIT=no
     AC_MSG_CHECKING([whether runnable 64 bit binaries are built by default])
     AC_RUN_IFELSE(int main(void) {return (sizeof(void*)*8==64)?0:1;},
-       DEFAULT_64BIT=yes, DEFAULT_64BIT=no, DEFAULT_64BIT=unknown)
+       DEFAULT_64BIT=yes, DEFAULT_64BIT=no, DEFAULT_64BIT=${ac_cv_64bit:unknown})
     BITS_GOT=unknown
     
     # 'OK' here means, we can exit any further checking, everything's copa
diff --git a/source/config/Makefile.inc.in b/source/config/Makefile.inc.in
index d7dfdbe..f5e63ec 100644
--- a/source/config/Makefile.inc.in
+++ b/source/config/Makefile.inc.in
@@ -110,7 +110,7 @@ ICULIBS_ICUIO = -l$(ICUPREFIX)io$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
 ICULIBS_OBSOLETE = -l$(ICUPREFIX)obsolete$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
 ICULIBS_LAYOUT = -l$(ICUPREFIX)le$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
 ICULIBS_LAYOUTEX = -l$(ICUPREFIX)lx$(ICULIBSUFFIX)$(ICULIBSUFFIX_VERSION)
-ICULIBS_BASE = $(LIBS) -L$(libdir)
+ICULIBS_BASE = $(LIBS)
 
 # for icu-config to test with
 ICULIBS_COMMON_LIB_NAME = ${LIBICU}uc${ICULIBSUFFIX}${ICULIBSUFFIX_VERSION}.${SO}
@@ -182,7 +182,7 @@ AR = @AR@
 CC = @CC@
 CFLAGS = @CFLAGS@ 
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@ $(LIBCPPFLAGS) -I$(prefix)/include
+CPPFLAGS = @CPPFLAGS@ $(LIBCPPFLAGS)
 CXX = @CXX@
 CXXFLAGS = @CXXFLAGS@ 
 DEFAULT_MODE = @DATA_PACKAGING_MODE@
diff --git a/source/configure b/source/configure
index beef9de..49cbb95 100755
--- a/source/configure
+++ b/source/configure
@@ -4031,7 +4031,7 @@ $as_echo "$as_me: error: Don't specify bitness when cross compiling. See readme.
     { $as_echo "$as_me:$LINENO: checking whether runnable 64 bit binaries are built by default" >&5
 $as_echo_n "checking whether runnable 64 bit binaries are built by default... " >&6; }
     if test "$cross_compiling" = yes; then
-  DEFAULT_64BIT=unknown
+  DEFAULT_64BIT=${ac_cv_64bit:unknown}
 else
   cat >conftest.$ac_ext <<_ACEOF
 int main(void) {return (sizeof(void*)*8==64)?0:1;}
