diff --git a/Makefile b/wsapi-1.3.4/Makefile
index ec6a3c2..83fc062 100644
--- a/Makefile
+++ b/Makefile
@@ -17,13 +17,14 @@ src/fastcgi/lfcgi.so: src/fastcgi/lfcgi.o src/fastcgi/lfcgi.h
 	$(CC) $(CFLAGS) $(LIB_OPTION) -o src/fastcgi/lfcgi.so src/fastcgi/lfcgi.o -lfcgi 
 
 install:
-	mkdir -p $(LUA_DIR)/wsapi
+	mkdir -p $(LUA_DIR)/wsapi $(BIN_DIR)/
 	cp src/wsapi/*.lua $(LUA_DIR)/wsapi
 	cp src/launcher/wsapi.cgi $(BIN_DIR)/
-	cp src/launcher/wsapi.fcgi $(BIN_DIR)/
 
 install-fcgi:
+	mkdir -p $(LUA_LIBDIR)/ $(BIN_DIR)/
 	cp src/fastcgi/lfcgi.so $(LUA_LIBDIR)/
+	cp src/launcher/wsapi.fcgi $(BIN_DIR)/
 
 install-rocks: install
 	mkdir -p $(PREFIX)/samples
