diff --git a/alsamixer/Makefile.am b/alsamixer/Makefile.am
index 1de47c6..40593f1 100644
--- a/alsamixer/Makefile.am
+++ b/alsamixer/Makefile.am
@@ -1,4 +1,5 @@
 AM_CFLAGS = @CURSES_CFLAGS@ -DCURSESINC="@CURSESINC@"
+INCLUDES = -I$(top_srcdir)/include
 LDADD = @CURSESLIB@
 
 bin_PROGRAMS = alsamixer
diff --git a/alsamixer/Makefile.in b/alsamixer/Makefile.in
index 8455506..f91df44 100644
--- a/alsamixer/Makefile.in
+++ b/alsamixer/Makefile.in
@@ -61,7 +61,7 @@ am_alsamixer_OBJECTS = card_select.$(OBJEXT) cli.$(OBJEXT) \
 alsamixer_OBJECTS = $(am_alsamixer_OBJECTS)
 alsamixer_LDADD = $(LDADD)
 alsamixer_DEPENDENCIES =
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include -I$(top_srcdir)/include
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
