diff --git a/networking/tcpudp.c b/networking/tcpudp.c
index b532e43..b7e5841 100644
--- a/networking/tcpudp.c
+++ b/networking/tcpudp.c
@@ -30,14 +30,6 @@
  */
 
 #include "libbb.h"
-
-/* Wants <limits.h> etc, thus included after libbb.h: */
-#ifdef __linux__
-#include <linux/types.h> /* for __be32 etc */
-#include <linux/netfilter_ipv4.h>
-#endif
-
-// TODO: move into this file:
 #include "tcpudp_perhost.h"
 
 #ifdef SSLSVD
