diff --git a/Makefile.am b/Makefile.am
index fa6ef15..11d311e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -14,7 +14,7 @@ Xtransinclude_HEADERS = \
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xtrans.m4
 
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xtrans.pc
 
 MAINTAINERCLEANFILES = ChangeLog INSTALL
diff --git a/Makefile.in b/Makefile.in
index 04324ad..bc4f294 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -272,7 +272,7 @@ Xtransinclude_HEADERS = \
 
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xtrans.m4
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xtrans.pc
 MAINTAINERCLEANFILES = ChangeLog INSTALL
 EXTRA_DIST = ${aclocal_DATA}
