diff --git a/Makefile.am b/Makefile.am
index fa6ef15..11d311e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -12,7 +12,7 @@ Xtransinclude_HEADERS = \
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xtrans.m4
 
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xtrans.pc
 
 EXTRA_DIST = xtrans.pc.in ${aclocal_DATA} ChangeLog Xtrans.mm
diff --git a/Makefile.in b/Makefile.in
index 04324ad..bc4f294 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -222,7 +222,7 @@ Xtransinclude_HEADERS = \
 
 aclocaldir = $(datadir)/aclocal
 aclocal_DATA = xtrans.m4
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = xtrans.pc
 EXTRA_DIST = xtrans.pc.in ${aclocal_DATA} ChangeLog Xtrans.mm
 MAINTAINERCLEANFILES = ChangeLog
