diff --git a/Makefile.am b/Makefile.am
index 3db4e37..303730e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,11 @@
 
 AUTOMAKE_OPTIONS = 1.9 gnu dist-bzip2 check-news
 
-SUBDIRS = libpam tests libpamc libpam_misc modules po conf doc examples xtests
+SUBDIRS = libpam tests libpamc libpam_misc modules po conf examples xtests
+
+if ENABLE_DOCUMENTATION
+SUBDIRS += doc
+endif
 
 CLEANFILES = *~
 
diff --git a/configure.in b/configure.in
index ae762a2..869d15b 100644
--- a/configure.in
+++ b/configure.in
@@ -531,33 +531,44 @@ AC_CHECK_FUNCS(inet_ntop inet_pton innetgr ruserok_af)
 AC_CHECK_FUNCS(unshare, [UNSHARE=yes], [UNSHARE=no])
 AM_CONDITIONAL([HAVE_UNSHARE], [test "$UNSHARE" = yes])
 
-AC_ARG_ENABLE([regenerate-docu],
-  AC_HELP_STRING([--disable-regenerate-docu], [Don't re-build documentation from XML souces]),
+AC_ARG_ENABLE([documentation],
+  AC_HELP_STRING([--disable-documentation], ["Don't build documentation at all"]),
   [enable_docu=$enableval], [enable_docu=yes])
+
+if test "x$enable_docu" != "xno"; then
+
+AC_ARG_ENABLE([regenerate-docu],
+  AC_HELP_STRING([--disable-regenerate-docu], ["Don't re-build documentation from XML souces"]),
+  [enable_xml_docu=$enableval], [enable_xml_docu=yes])
 dnl
 dnl Check for xsltproc
 dnl
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 if test -z "$XSLTPROC"; then
-     enable_docu=no
+     enable_xml_docu=no
 fi
 AC_PATH_PROG([XMLLINT], [xmllint],[/bin/true])
 dnl check for DocBook DTD and stylesheets in the local catalog.
 JH_CHECK_XML_CATALOG([-//OASIS//DTD DocBook XML V4.4//EN],
-                [DocBook XML DTD V4.4], [], enable_docu=no)
+                [DocBook XML DTD V4.4], [], enable_xml_docu=no)
 JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl],
-                [DocBook XSL Stylesheets], [], enable_docu=no)
+                [DocBook XSL Stylesheets], [], enable_xml_docu=no)
 
 AC_PATH_PROG([BROWSER], [w3m])
 if test ! -z "$BROWSER"; then
      BROWSER="$BROWSER -T text/html -dump"
 else
-     enable_docu=no
+     enable_xml_docu=no
 fi
 
 AC_PATH_PROG([FO2PDF], [fop])
 
-AM_CONDITIONAL(ENABLE_REGENERATE_MAN, test x$enable_docu != xno)
+else
+     enable_xml_docu=no
+fi
+
+AM_CONDITIONAL(ENABLE_DOCUMENTATION, test x$enable_docu != xno)
+AM_CONDITIONAL(ENABLE_REGENERATE_MAN, test x$enable_xml_docu != xno)
 AM_CONDITIONAL(ENABLE_GENERATE_PDF, test ! -z "$FO2PDF")
 
 
