diff --git a/Makefile b/Makefile
index 5637c2b..f519ce7 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,6 @@ DIST_SRC_TGZ = glew-$(GLEW_VERSION).tgz
 
 AR = ar
 INSTALL = install
-STRIP = strip
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
@@ -182,10 +181,8 @@ install.lib: glew.lib
 # runtime
 ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(STRIP) -x lib/$(LIB.SHARED)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(BINDIR)/
 else
-	$(STRIP) -x lib/$(LIB.SHARED)
 	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
 endif
@@ -195,7 +192,6 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 else
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
 endif
-	$(STRIP) -x lib/$(LIB.STATIC)
 	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 
 install.lib.mx: glew.lib.mx
@@ -203,11 +199,9 @@ install.lib.mx: glew.lib.mx
 # runtime
 ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(STRIP) -x lib/$(LIB.SHARED.MX)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(BINDIR)/
 else
-	$(STRIP) -x lib/$(LIB.SHARED.MX)
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED.MX) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(LIBDIR)/
 	$(LN) $(LIB.SHARED.MX) $(LIBDIR)/$(LIB.SONAME.MX)
 endif
 # development files
@@ -216,7 +210,6 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 else
 	$(LN) $(LIB.SHARED.MX) $(LIBDIR)/$(LIB.DEVLNK.MX)
 endif
-	$(STRIP) -x lib/$(LIB.STATIC.MX)
 	$(INSTALL) -m 0644 lib/$(LIB.STATIC.MX) $(LIBDIR)/
 
 install.bin: glew.bin
