diff --git a/linux/arm/syscallent.h b/linux/arm/syscallent.h
index a8ec923..490c485 100644
--- a/linux/arm/syscallent.h
+++ b/linux/arm/syscallent.h
@@ -430,7 +430,7 @@
 	{ 5,	0,	printargs,		"SYS_397"	}, /* 397 */
 	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
 	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
-
+#if !defined(__ARM_EABI__)
 #if SYS_socket_subcall != 400
  #error fix me
 #endif
@@ -452,7 +452,9 @@
 	{ 5,	TN,	sys_getsockopt,		"getsockopt"	}, /* 415 */
 	{ 5,	TN,	sys_sendmsg,		"sendmsg"	}, /* 416 */
 	{ 5,	TN,	sys_recvmsg,		"recvmsg"	}, /* 417 */
+#endif /* !__ARM_EABI__ */
 
+#if !defined(__ARM_EABI__)
 #if SYS_ipc_subcall != 418
  #error fix me
 #endif
@@ -481,3 +483,4 @@
 	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
 	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
 	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
+#endif /* !__ARM_EABI__ */
