diff --git a/configure.ac b/configure.ac
index 2822ebc..312f430 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,10 +17,6 @@ AM_INIT_AUTOMAKE([foreign -Wall -Werror])
 AM_MAINTAINER_MODE
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES(yes)])
 
-# Configure libtool
-#LT_PREREQ([2.2])
-LT_INIT
-
 #######################
 ### Checks
 
@@ -33,6 +29,9 @@ AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_PROG_CXX
 AM_PROG_CC_C_O
+# automake 1.12 seems to require this, but automake 1.11 doesn't recognize it
+m4_pattern_allow([AM_PROG_AR])
+AM_PROG_AR
 AC_PROG_LIBTOOL
 
 # Check some compiler flags
