--- a/src/priv.c	2012-02-29 22:56:03.000000000 +0100
+++ b/src/priv.c	2012-09-17 11:31:44.717126873 +0200
@@ -245,7 +245,7 @@
 	if (uname(&un) != 0)
 		fatal("[priv]: failed to get system information");
 	if ((hp = gethostbyname(un.nodename)) == NULL) {
-		LLOG_INFO("[priv]: unable to get system name");
+		LLOG_DEBUG("[priv]: unable to get system name");
 		res_init();
                 len = strlen(un.nodename);
                 must_write(remote, &len, sizeof(int));
