diff --git a/configure.ac b/configure.ac
index 043ab5f..8f59794 100755
--- a/configure.ac
+++ b/configure.ac
@@ -48,6 +48,9 @@ if test x"$CC_FOR_BUILD" = x; then
 	fi
 fi
 AC_SUBST([CC_FOR_BUILD])
+AC_SUBST([CPPFLAGS_FOR_BUILD])
+AC_SUBST([CFLAGS_FOR_BUILD])
+AC_SUBST([LDFLAGS_FOR_BUILD])
 
 PKG_CHECK_MODULES(XT, sm x11 xproto kbproto)
 
diff --git a/util/Makefile.am b/util/Makefile.am
index 0d3ff01..5060801 100644
--- a/util/Makefile.am
+++ b/util/Makefile.am
@@ -5,6 +5,9 @@ noinst_PROGRAMS = makestrs
 endif
 
 CC = @CC_FOR_BUILD@
+CPPFLAGS = @CPPFLAGS_FOR_BUILD@
+CFLAGS = @CFLAGS_FOR_BUILD@
+LDFLAGS = @LDFLAGS_FOR_BUILD@
 
 EXTRA_DIST = \
 	Shell.ht \
