diff --git a/Makefile.am b/Makefile.am
index 52e83c2..3341478 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -6,9 +6,8 @@ nodm_SOURCES = nodm.c
 nodm_LDFLAGS = $(PAM_LIBS)
 nodm_CFLAGS = -DNODM_SESSION='"$(sbindir)/nodm"'
 
-nodm.8: nodm
-	help2man --section=8 --name="X display manager for automatic logins" \
-		 --no-info --include=nodm-man-extras ./$< > $@
+nodm.8: nodm.8_shipped
+	cp $< $@
 
 man_MANS = nodm.8
 
