diff --git a/src/util/Makefile.am b/src/util/Makefile.am
index 3ff421d..ac4abb0 100644
--- a/src/util/Makefile.am
+++ b/src/util/Makefile.am
@@ -4,7 +4,9 @@ noinst_PROGRAMS=makekeys
 
 makekeys_CFLAGS=$(X11_CFLAGS)
 
-#override CC = gcc
+override CC      = gcc
+override CFLAGS  =
+override LDFLAGS =
 LINK = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
 EXTRA_DIST = mkks.sh
diff --git a/src/util/Makefile.in b/src/util/Makefile.in
index dc78625..2a3042e 100644
--- a/src/util/Makefile.in
+++ b/src/util/Makefile.in
@@ -227,7 +227,9 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 makekeys_CFLAGS = $(X11_CFLAGS)
 
-#override CC = gcc
+override CC      = gcc
+override CFLAGS  =
+override LDFLAGS =
 LINK = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 EXTRA_DIST = mkks.sh
 
