diff --git a/configure b/configure
index a680f65..303a191 100755
--- a/configure
+++ b/configure
@@ -818,6 +818,9 @@ PERL
 RAWCPPFLAGS
 RAWCPP
 CC_FOR_BUILD
+CFLAGS_FOR_BUILD
+LD_FOR_BUILD
+LDFLAGS_FOR_BUILD
 LTCXXCOMPILE
 LTCOMPILE
 DOLT_BASH
@@ -20863,8 +20866,64 @@ fi
   test -n "$CC_FOR_BUILD" && break
 done
 
+	       CFLAGS_FOR_BUILD="-O2 -g"
 	else
 	       CC_FOR_BUILD="$CC"
+	       CFLAGS_FOR_BUILD="$CFLAGS"
+	fi
+fi
+
+
+
+
+if test x"$LD_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       for ac_prog in ld
+do
+  # Extract the first word of "$ac_prog", so it can be a program name with args.
+set dummy $ac_prog; ac_word=$2
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
+$as_echo_n "checking for $ac_word... " >&6; }
+if test "${ac_cv_prog_LD_FOR_BUILD+set}" = set; then :
+  $as_echo_n "(cached) " >&6
+else
+  if test -n "$LD_FOR_BUILD"; then
+  ac_cv_prog_LD_FOR_BUILD="$LD_FOR_BUILD" # Let the user override the test.
+else
+as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+for as_dir in $PATH
+do
+  IFS=$as_save_IFS
+  test -z "$as_dir" && as_dir=.
+    for ac_exec_ext in '' $ac_executable_extensions; do
+  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_test_x "$as_dir/$ac_word$ac_exec_ext"; }; then
+    ac_cv_prog_LD_FOR_BUILD="$ac_prog"
+    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
+    break 2
+  fi
+done
+  done
+IFS=$as_save_IFS
+
+fi
+fi
+LD_FOR_BUILD=$ac_cv_prog_LD_FOR_BUILD
+if test -n "$LD_FOR_BUILD"; then
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $LD_FOR_BUILD" >&5
+$as_echo "$LD_FOR_BUILD" >&6; }
+else
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+fi
+
+
+  test -n "$LD_FOR_BUILD" && break
+done
+
+	       LDFLAGS_FOR_BUILD="-O2 -g"
+	else
+	       LD_FOR_BUILD="$LD"
+	       LDFLAGS_FOR_BUILD="$LDFLAGS"
 	fi
 fi
 
diff --git a/configure.ac b/configure.ac
index 0eea575..8fec648 100644
--- a/configure.ac
+++ b/configure.ac
@@ -33,11 +33,26 @@ AC_PROG_CC
 if test x"$CC_FOR_BUILD" = x; then
 	if test x"$cross_compiling" = xyes; then
 	       AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+	       CFLAGS_FOR_BUILD="-O2 -g"
 	else
 	       CC_FOR_BUILD="$CC"
+	       CFLAGS_FOR_BUILD="$CFLAGS"
 	fi
 fi
 AC_SUBST([CC_FOR_BUILD])
+AC_SUBST([CFLAGS_FOR_BUILD])
+
+if test x"$LD_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+		AC_CHECK_PROGS(LD_FOR_BUILD, ld)
+		LDFLAGS_FOR_BUILD=""
+	else
+		LD_FOR_BUILD="$LD"
+		LDFLAGS_FOR_BUILD="$LDFLAGS"
+	fi
+fi
+AC_SUBST([LD_FOR_BUILD])
+AC_SUBST([LDFLAGS_FOR_BUILD])
 
 XORG_PROG_RAWCPP
 
diff --git a/src/util/Makefile.am b/src/util/Makefile.am
index 7fc8da5..4632156 100644
--- a/src/util/Makefile.am
+++ b/src/util/Makefile.am
@@ -4,6 +4,9 @@ noinst_PROGRAMS=makekeys
 makekeys_CFLAGS=$(X11_CFLAGS)
 
 CC = @CC_FOR_BUILD@
+CFLAGS = @CFLAGS_FOR_BUILD@
+LD = @LD_FOR_BUILD@
+LDFLAGS = @LDFLAGS_FOR_BUILD@
 
 EXTRA_DIST = mkks.sh
 
diff --git a/src/util/Makefile.in b/src/util/Makefile.in
index 6fee791..3c08531 100644
--- a/src/util/Makefile.in
+++ b/src/util/Makefile.in
@@ -101,7 +101,7 @@ BIGFONT_LIBS = @BIGFONT_LIBS@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS_FOR_BUILD@
 CHANGELOG_CMD = @CHANGELOG_CMD@
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
@@ -138,7 +138,8 @@ INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 KEYSYMDEF = @KEYSYMDEF@
 LAUNCHD = @LAUNCHD@
-LDFLAGS = @LDFLAGS@
+LD = @LD@
+LDFLAGS = @LDFLAGS_FOR_BUILD@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
