diff --git a/configure.ac b/configure.ac
index ab6d5a3..8a45ffe 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,6 +37,9 @@ if test x"$CC_FOR_BUILD" = x; then
 	fi
 fi
 AC_SUBST([CC_FOR_BUILD])
+AC_SUBST([CPPFLAGS_FOR_BUILD])
+AC_SUBST([CFLAGS_FOR_BUILD])
+AC_SUBST([LDFLAGS_FOR_BUILD])
 
 XORG_PROG_RAWCPP
 
diff --git a/src/util/Makefile.am b/src/util/Makefile.am
index c9e81c0..ccf1f9d 100644
--- a/src/util/Makefile.am
+++ b/src/util/Makefile.am
@@ -5,6 +5,9 @@ noinst_PROGRAMS=makekeys
 makekeys_CFLAGS=$(X11_CFLAGS)
 
 CC = @CC_FOR_BUILD@
+CPPFLAGS = @CPPFLAGS_FOR_BUILD@
+CFLAGS = @CFLAGS_FOR_BUILD@
+LDFLAGS = @LDFLAGS_FOR_BUILD@
 
 EXTRA_DIST = mkks.sh
 
