--- a/configure.ac	2007-06-27 15:40:31.000000000 +0200
+++ b/configure.ac	2007-06-27 15:41:02.000000000 +0200
@@ -214,7 +214,7 @@
 		*-I*)
 			dir=`echo "$flag" | sed 's/ *-I//'`
 			file="$dir/X11/keysymdef.h"
-			if test -f "$file"; then
+			if test -f "$file" -o -f "${DESTDIR}$file"; then
 				KEYSYMDEF="$file"
 			fi
 			;;
--- a/configure	2007-06-27 15:40:31.000000000 +0200
+++ b/configure	2007-06-27 15:41:02.000000000 +0200
@@ -22767,7 +22767,7 @@
 		*-I*)
 			dir=`echo "$flag" | sed 's/ *-I//'`
 			file="$dir/X11/keysymdef.h"
-			if test -f "$file"; then
+			if test -f "$file" -o -f "${DESTDIR}$file"; then
 				KEYSYMDEF="$file"
 			fi
 			;;
