diff --git a/configure.ac b/configure.ac
index 8ad1f00..ef0af06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -308,16 +308,16 @@ AC_MSG_CHECKING([keysym definitions])
 AC_ARG_WITH([keysymdefdir],
         [AC_HELP_STRING([--with-keysymdefdir=DIR], [The location of keysymdef.h (defaults to xproto include dir)])],
         [KEYSYMDEFDIR=$withval],
-        [KEYSYMDEFDIR=`$PKG_CONFIG --variable=includedir xproto`/X11])
+        [KEYSYMDEFDIR=`$PKG_CONFIG --variable=includex11dir xproto`])
 
-if test ! -d "$KEYSYMDEFDIR"; then
+if test ! -d "$PKG_CONFIG_SYSROOT_DIR$KEYSYMDEFDIR"; then
 	AC_MSG_ERROR([$KEYSYMDEFDIR doesn't exist or isn't a directory])
 fi
 
 FILES="keysymdef.h XF86keysym.h Sunkeysym.h DECkeysym.h HPkeysym.h"
 for i in $FILES; do
-    if test -f "$KEYSYMDEFDIR/$i"; then
-            KEYSYMDEFS="$KEYSYMDEFS $KEYSYMDEFDIR/$i"
+    if test -f "$PKG_CONFIG_SYSROOT_DIR$KEYSYMDEFDIR/$i"; then
+            KEYSYMDEFS="$KEYSYMDEFS $PKG_CONFIG_SYSROOT_DIR$KEYSYMDEFDIR/$i"
     elif test "x$i" = "xkeysymdef.h"; then
 	    AC_MSG_ERROR([Cannot find keysymdef.h])
     fi
