diff --git a/extras/ata_id/ata_id.c b/extras/ata_id/ata_id.c
index 25c7710..bfb7caf 100644
--- a/extras/ata_id/ata_id.c
+++ b/extras/ata_id/ata_id.c
@@ -44,6 +44,10 @@
 #include "libudev.h"
 #include "libudev-private.h"
 
+#if defined(__UCLIBC__)
+#define le16toh(x) __le16_to_cpu(x)
+#endif
+
 #define COMMAND_TIMEOUT 2000
 
 /* Sends a SCSI command block */
diff --git a/libudev/libudev-monitor.c b/libudev/libudev-monitor.c
index 96c153f..adf6a61 100644
--- a/libudev/libudev-monitor.c
+++ b/libudev/libudev-monitor.c
@@ -24,6 +24,10 @@
 #include <linux/netlink.h>
 #include <linux/filter.h>
 
+#ifndef SOCK_CLOEXEC
+#define SOCK_CLOEXEC 02000000
+#endif
+
 #include "libudev.h"
 #include "libudev-private.h"
 
diff --git a/udev/udevd.c b/udev/udevd.c
index fe3352c..21ebce8 100644
--- a/udev/udevd.c
+++ b/udev/udevd.c
@@ -44,6 +44,10 @@
 #include <sys/inotify.h>
 #include <sys/utsname.h>
 
+#ifndef SOCK_CLOEXEC
+#define SOCK_CLOEXEC 02000000
+#endif
+
 #include "udev.h"
 
 #define UDEVD_PRIORITY			-4
