diff --git a/configure b/configure
index 8843435..41838b5 100755
--- a/configure
+++ b/configure
@@ -40,6 +40,7 @@ for x; do
 	--mandir) MANDIR=$var;;
 	--with-ccopts|CFLAGS) CFLAGS=$var;;
 	CPPFLAGS) CPPFLAGS=$var;;
+	LDFLAGS) LDFLAGS=$var;;
 	--with-hook) HOOKSCRIPTS="$HOOKSCRIPTS${HOOKSCRIPTS:+ }$var";;
 	--with-hooks|HOOKSCRIPTS) HOOKSCRIPTS=$var; HOOKSET=true;;
 	--build) BUILD=$var;;
@@ -262,8 +266,8 @@ if [ -e "$LDELF" ]; then
 fi
 
 # Add CPPFLAGS and CFLAGS to CC for testing features
-XCC="$CC `$SED -n -e 's/CPPLAGS+=*\(.*\)/\1/p' $CONFIG_MK`"
-XCC="$XCC `$SED -n -e 's/CFLAGS+=*\(.*\)/\1/p' $CONFIG_MK`"
+XCC="$CC `$SED -n -e 's/CPPLAGS+\?=\s*\(.*\)/\1/p' $CONFIG_MK`"
+XCC="$XCC `$SED -n -e 's/CFLAGS+\?=\s*\(.*\)/\1/p' $CONFIG_MK`"
 
 abort=false
 # We require the libc to support non standard functions, like getifaddrs
