diff --git a/pango/Makefile.am b/pango/Makefile.am
index 9b31610..e91c78e 100644
--- a/pango/Makefile.am
+++ b/pango/Makefile.am
@@ -125,10 +125,11 @@ pangoinclude_HEADERS =	   \
 	pango-features.h   \
 	pango-enum-types.h
 
+GLIB_MKENUMS ?= glib-mkenums
 pango-enum-types.h: s-enum-types-h
 	@true
 s-enum-types-h: @REBUILD@ $(pango_headers) Makefile
-	( cd $(srcdir) && glib-mkenums \
+	( cd $(srcdir) && $(GLIB_MKENUMS) \
 			--fhead "#ifndef __PANGO_ENUM_TYPES_H__\n#define __PANGO_ENUM_TYPES_H__\n\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
 			--fprod "/* enumerations from \"@filename@\" */\n" \
 			--vhead "GType @enum_name@_get_type (void);\n#define @ENUMPREFIX@_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n" 	\
diff --git a/pango/Makefile.in b/pango/Makefile.in
index f5afafd..a353d3f 100644
--- a/pango/Makefile.in
+++ b/pango/Makefile.in
@@ -1400,10 +1399,11 @@ uninstall-am: uninstall-aliasDATA uninstall-binPROGRAMS \
 pango-win32-res.o: pango.rc
 	$(WINDRES) $< $@
 
+GLIB_MKENUMS ?= glib-mkenums
 pango-enum-types.h: s-enum-types-h
 	@true
 s-enum-types-h: @REBUILD@ $(pango_headers) Makefile
-	( cd $(srcdir) && glib-mkenums \
+	( cd $(srcdir) && $(GLIB_MKENUMS) \
 			--fhead "#ifndef __PANGO_ENUM_TYPES_H__\n#define __PANGO_ENUM_TYPES_H__\n\n#include <glib-object.h>\n\nG_BEGIN_DECLS\n" \
 			--fprod "/* enumerations from \"@filename@\" */\n" \
 			--vhead "GType @enum_name@_get_type (void);\n#define @ENUMPREFIX@_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n" 	\
