diff --git a/main.c b/main.c
index 1dc973d..a823a39 100644
--- a/main.c
+++ b/main.c
@@ -2482,7 +2482,7 @@ get_pty(int *pty, char *from GCC_UNUSED)
 #ifdef __GLIBC__		/* if __GLIBC__ and USE_USG_PTYS, we know glibc >= 2.1 */
     /* GNU libc 2 allows us to abstract away from having to know the
        master pty device name. */
-    if ((*pty = getpt()) >= 0) {
+    if ((*pty = posix_openpt(O_RDWR | O_NOCTTY)) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {	/* if filesystem is trashed, this may be null */
 	    strcpy(ttydev, name);
