diff --git a/test/basic/Makefile.am b/test/basic/Makefile.am
index 169e193..8f0e312 100644
--- a/test/basic/Makefile.am
+++ b/test/basic/Makefile.am
@@ -23,7 +23,7 @@
 noinst_PROGRAMS = test_01 test_02 test_03 test_04 test_05 test_06 \
 		test_07 test_08 test_09 test_10 test_11
 
-AM_CFLAGS = -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
+AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/va -I$(top_srcdir)/src/x11
 
 TEST_LIBS = $(top_builddir)/va/$(libvabackendlib) $(top_builddir)/va/$(libvacorelib) -lpthread -lX11
 
