From 524875a847129e7ce3ce093a5821bef57d5a9a86 Mon Sep 17 00:00:00 2001
From: Dirk Leber <dirk.leber@avionic-design.de>
Date: Mon, 26 Jan 2015 09:56:14 +0100
Subject: [PATCH 2/7] gles2: Raise priority to force default usage of plugin

Signed-off-by: Dirk Leber <dirk.leber@avionic-design.de>
---
 modules/video_output/gl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/video_output/gl.c b/modules/video_output/gl.c
index 392ffa1..6bf01b1 100644
--- a/modules/video_output/gl.c
+++ b/modules/video_output/gl.c
@@ -49,7 +49,7 @@ vlc_module_begin ()
 # define MODULE_VARNAME "gles2"
     set_shortname (N_("OpenGL ES2"))
     set_description (N_("OpenGL for Embedded Systems 2 video output"))
-    set_capability ("vout display", /*265*/0)
+    set_capability ("vout display", /*265*/500)
     set_callbacks (Open, Close)
     add_shortcut ("opengles2", "gles2")
     add_module ("gles2", "opengl es2", NULL,
-- 
2.1.0

