diff --git a/localedata/Makefile b/localedata/Makefile
index 57cfd52..cdfbcce 100644
--- a/localedata/Makefile
+++ b/localedata/Makefile
@@ -192,9 +192,11 @@ include SUPPORTED
 
 INSTALL-SUPPORTED-LOCALES=$(addprefix install-, $(SUPPORTED-LOCALES))
 
+if-exists = $(if $(wildcard $(1)),$(1) $(2))
+
 # Sometimes the whole collection of locale files should be installed.
 LOCALEDEF=I18NPATH=. GCONV_PATH=$(common-objpfx)iconvdata LC_ALL=C \
-$(common-objpfx)elf/ld.so --library-path $(rpath-link) $(common-objpfx)locale/localedef
+$(call if-exists,$(common-objpfx)elf/ld.so, --library-path $(rpath-link)) $(common-objpfx)locale/localedef
 install-locales: $(INSTALL-SUPPORTED-LOCALES)
 
 install-locales-dir:
