diff --git a/Makefile.shared b/Makefile.shared
index 3569832..89a2156 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -7,7 +7,7 @@
 
 # CC contains the current compiler.  This one MUST be defined
 CC=cc
-CFLAGS=$(CFLAG)
+CFLAGS=$(EXTRA_CPPFLAGS) $(CFLAG) $(EXTRA_CFLAGS)
 # LDFLAGS contains flags to be used when temporary object files (when building
 # shared libraries) are created, or when an application is linked.
 # SHARED_LDFLAGS contains flags to be used when the shared library is created.
diff --git a/apps/Makefile b/apps/Makefile
index fa32d2d..ad8fff9 100644
--- a/apps/Makefile
+++ b/apps/Makefile
@@ -20,7 +20,7 @@ EXE_EXT=
 
 SHLIB_TARGET=
 
-CFLAGS= -DMONOLITH $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) -DMONOLITH $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile makeapps.com install.com
 
diff --git a/crypto/Makefile b/crypto/Makefile
index c1033f6..9295086 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -23,7 +23,7 @@ RECURSIVE_MAKE=	[ -n "$(SDIRS)" ] && for i in $(SDIRS) ; do \
 PEX_LIBS=
 EX_LIBS=
  
-CFLAGS= $(INCLUDE) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDE) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDE) $(ASFLAG)
 AFLAGS=$(ASFLAGS)
 CPUID_OBJ=mem_clr.o
diff --git a/crypto/aes/Makefile b/crypto/aes/Makefile
index c501a43..8e0dd51 100644
--- a/crypto/aes/Makefile
+++ b/crypto/aes/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 AES_ENC=aes_core.o aes_cbc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/asn1/Makefile b/crypto/asn1/Makefile
index 160544e..506ea5c 100644
--- a/crypto/asn1/Makefile
+++ b/crypto/asn1/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/bf/Makefile b/crypto/bf/Makefile
index dd2c2c7..966f360 100644
--- a/crypto/bf/Makefile
+++ b/crypto/bf/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 BF_ENC=		bf_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/bio/Makefile b/crypto/bio/Makefile
index c395d80..f37aa1b 100644
--- a/crypto/bio/Makefile
+++ b/crypto/bio/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/bn/Makefile b/crypto/bn/Makefile
index aabc4f5..75f536c 100644
--- a/crypto/bn/Makefile
+++ b/crypto/bn/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 BN_ASM=		bn_asm.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/buffer/Makefile b/crypto/buffer/Makefile
index 9f3a88d..b410f0d 100644
--- a/crypto/buffer/Makefile
+++ b/crypto/buffer/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/camellia/Makefile b/crypto/camellia/Makefile
index ff5fe4a..e49284d 100644
--- a/crypto/camellia/Makefile
+++ b/crypto/camellia/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 CMLL_ENC= camellia.o cmll_misc.o cmll_cbc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/cast/Makefile b/crypto/cast/Makefile
index 0acc38f..9fb36e9 100644
--- a/crypto/cast/Makefile
+++ b/crypto/cast/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 CAST_ENC=c_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/cms/Makefile b/crypto/cms/Makefile
index 5837049..531e8be 100644
--- a/crypto/cms/Makefile
+++ b/crypto/cms/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/comp/Makefile b/crypto/comp/Makefile
index efda832..8f80818 100644
--- a/crypto/comp/Makefile
+++ b/crypto/comp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/conf/Makefile b/crypto/conf/Makefile
index 78bb324..c483106 100644
--- a/crypto/conf/Makefile
+++ b/crypto/conf/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/des/Makefile b/crypto/des/Makefile
index ae98226..e072e05 100644
--- a/crypto/des/Makefile
+++ b/crypto/des/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 RANLIB=		ranlib
 DES_ENC=	des_enc.o fcrypt_b.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/dh/Makefile b/crypto/dh/Makefile
index f23b4f7..15a9eef 100644
--- a/crypto/dh/Makefile
+++ b/crypto/dh/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= dhtest.c
diff --git a/crypto/dsa/Makefile b/crypto/dsa/Makefile
index 8073c4e..b87cf20 100644
--- a/crypto/dsa/Makefile
+++ b/crypto/dsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=dsatest.c
diff --git a/crypto/dso/Makefile b/crypto/dso/Makefile
index fb2709e..0f0cec0 100644
--- a/crypto/dso/Makefile
+++ b/crypto/dso/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/ec/Makefile b/crypto/ec/Makefile
index db380ed..7e818ee 100644
--- a/crypto/ec/Makefile
+++ b/crypto/ec/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ectest.c
diff --git a/crypto/ecdh/Makefile b/crypto/ecdh/Makefile
index 65d8904..91a3cbc 100644
--- a/crypto/ecdh/Makefile
+++ b/crypto/ecdh/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g -Wall
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ecdhtest.c
diff --git a/crypto/ecdsa/Makefile b/crypto/ecdsa/Makefile
index e89e0c0..8658954 100644
--- a/crypto/ecdsa/Makefile
+++ b/crypto/ecdsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g -Wall
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ecdsatest.c
diff --git a/crypto/engine/Makefile b/crypto/engine/Makefile
index 9c21482..9269979 100644
--- a/crypto/engine/Makefile
+++ b/crypto/engine/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= enginetest.c
diff --git a/crypto/err/Makefile b/crypto/err/Makefile
index 862b23b..38fefbb 100644
--- a/crypto/err/Makefile
+++ b/crypto/err/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/evp/Makefile b/crypto/evp/Makefile
index 82825e5..3ccd510 100644
--- a/crypto/evp/Makefile
+++ b/crypto/evp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=evp_test.c
diff --git a/crypto/hmac/Makefile b/crypto/hmac/Makefile
index 0e91709..1464180 100644
--- a/crypto/hmac/Makefile
+++ b/crypto/hmac/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=hmactest.c
diff --git a/crypto/idea/Makefile b/crypto/idea/Makefile
index b2e7add..79b7feb 100644
--- a/crypto/idea/Makefile
+++ b/crypto/idea/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=ideatest.c
diff --git a/crypto/jpake/Makefile b/crypto/jpake/Makefile
index 110c49c..44b6369 100644
--- a/crypto/jpake/Makefile
+++ b/crypto/jpake/Makefile
@@ -1,7 +1,7 @@
 DIR=jpake
 TOP=../..
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 LIB=$(TOP)/libcrypto.a
 LIBOBJ=jpake.o jpake_err.o
diff --git a/crypto/krb5/Makefile b/crypto/krb5/Makefile
index 1407739..a948c73 100644
--- a/crypto/krb5/Makefile
+++ b/crypto/krb5/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/lhash/Makefile b/crypto/lhash/Makefile
index 82bddac..57fb377 100644
--- a/crypto/lhash/Makefile
+++ b/crypto/lhash/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/md2/Makefile b/crypto/md2/Makefile
index 17f878a..50ff27a 100644
--- a/crypto/md2/Makefile
+++ b/crypto/md2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=md2test.c
diff --git a/crypto/md4/Makefile b/crypto/md4/Makefile
index c94a139..8f5e10d 100644
--- a/crypto/md4/Makefile
+++ b/crypto/md4/Makefile
@@ -11,7 +11,7 @@ CFLAG=-g
 MAKEFILE=       Makefile
 AR=             ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=md4test.c
diff --git a/crypto/md5/Makefile b/crypto/md5/Makefile
index 9858d53..b9f52be 100644
--- a/crypto/md5/Makefile
+++ b/crypto/md5/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 MD5_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/mdc2/Makefile b/crypto/mdc2/Makefile
index 1d064f1..9f28cfd 100644
--- a/crypto/mdc2/Makefile
+++ b/crypto/mdc2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= mdc2test.c
diff --git a/crypto/modes/Makefile b/crypto/modes/Makefile
index 6c85861..7a7624a 100644
--- a/crypto/modes/Makefile
+++ b/crypto/modes/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/objects/Makefile b/crypto/objects/Makefile
index a8aedbd..6999ac2 100644
--- a/crypto/objects/Makefile
+++ b/crypto/objects/Makefile
@@ -11,7 +11,7 @@ MAKEFILE=	Makefile
 AR=		ar r
 PERL=		perl
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/ocsp/Makefile b/crypto/ocsp/Makefile
index 60c414c..c3a7f80 100644
--- a/crypto/ocsp/Makefile
+++ b/crypto/ocsp/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/pem/Makefile b/crypto/pem/Makefile
index 2cc7801..6cc6a1a 100644
--- a/crypto/pem/Makefile
+++ b/crypto/pem/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/pkcs12/Makefile b/crypto/pkcs12/Makefile
index 3a7498f..5b37f1d 100644
--- a/crypto/pkcs12/Makefile
+++ b/crypto/pkcs12/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/pkcs7/Makefile b/crypto/pkcs7/Makefile
index 56dc682..f57f711 100644
--- a/crypto/pkcs7/Makefile
+++ b/crypto/pkcs7/Makefile
@@ -12,8 +12,8 @@ AR=		ar r
 
 PEX_LIBS=
 EX_LIBS=
- 
-CFLAGS= $(INCLUDES) $(CFLAG)
+
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/pqueue/Makefile b/crypto/pqueue/Makefile
index fb36a0c..47570ac 100644
--- a/crypto/pqueue/Makefile
+++ b/crypto/pqueue/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/rand/Makefile b/crypto/rand/Makefile
index 27694aa..0a67679 100644
--- a/crypto/rand/Makefile
+++ b/crypto/rand/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST= randtest.c
diff --git a/crypto/rc2/Makefile b/crypto/rc2/Makefile
index 73eac34..2a866bc 100644
--- a/crypto/rc2/Makefile
+++ b/crypto/rc2/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=rc2test.c
diff --git a/crypto/rc4/Makefile b/crypto/rc4/Makefile
index 264451a..a3ee7f0 100644
--- a/crypto/rc4/Makefile
+++ b/crypto/rc4/Makefile
@@ -12,7 +12,7 @@ AR=		ar r
 
 RC4_ENC=rc4_enc.o rc4_skey.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/rc5/Makefile b/crypto/rc5/Makefile
index 8a8b00e..f855817 100644
--- a/crypto/rc5/Makefile
+++ b/crypto/rc5/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 RC5_ENC=		rc5_enc.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/ripemd/Makefile b/crypto/ripemd/Makefile
index d5b1067..7cd063e 100644
--- a/crypto/ripemd/Makefile
+++ b/crypto/ripemd/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 RIP_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/rsa/Makefile b/crypto/rsa/Makefile
index bb64223..77a488f 100644
--- a/crypto/rsa/Makefile
+++ b/crypto/rsa/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=rsa_test.c
diff --git a/crypto/seed/Makefile b/crypto/seed/Makefile
index 4bc55e4..24df7c8 100644
--- a/crypto/seed/Makefile
+++ b/crypto/seed/Makefile
@@ -11,7 +11,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/sha/Makefile b/crypto/sha/Makefile
index e6eccb0..72aae15 100644
--- a/crypto/sha/Makefile
+++ b/crypto/sha/Makefile
@@ -13,7 +13,7 @@ AR=             ar r
 
 SHA1_ASM_OBJ=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/stack/Makefile b/crypto/stack/Makefile
index 5327692..9a4d0f2 100644
--- a/crypto/stack/Makefile
+++ b/crypto/stack/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/store/Makefile b/crypto/store/Makefile
index 0dcfd78..588f6ed 100644
--- a/crypto/store/Makefile
+++ b/crypto/store/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 #TEST= storetest.c
diff --git a/crypto/ts/Makefile b/crypto/ts/Makefile
index ad29b67..33be92a 100644
--- a/crypto/ts/Makefile
+++ b/crypto/ts/Makefile
@@ -17,8 +17,8 @@ AR=		ar r
 
 PEX_LIBS=
 EX_LIBS=
- 
-CFLAGS= $(INCLUDES) $(CFLAG)
+
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL= Makefile
 TEST=
diff --git a/crypto/txt_db/Makefile b/crypto/txt_db/Makefile
index e6f3033..499d8fa 100644
--- a/crypto/txt_db/Makefile
+++ b/crypto/txt_db/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 TEST=
diff --git a/crypto/ui/Makefile b/crypto/ui/Makefile
index a685659..bd25fea 100644
--- a/crypto/ui/Makefile
+++ b/crypto/ui/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile
 #TEST= uitest.c
diff --git a/crypto/whrlpool/Makefile b/crypto/whrlpool/Makefile
index 566b996..9e201ec 100644
--- a/crypto/whrlpool/Makefile
+++ b/crypto/whrlpool/Makefile
@@ -13,7 +13,7 @@ AR=		ar r
 
 WP_ASM_OBJ=wp_block.o
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 ASFLAGS= $(INCLUDES) $(ASFLAG)
 AFLAGS= $(ASFLAGS)
 
diff --git a/crypto/x509/Makefile b/crypto/x509/Makefile
index 72c8227..ae1182a 100644
--- a/crypto/x509/Makefile
+++ b/crypto/x509/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/crypto/x509v3/Makefile b/crypto/x509v3/Makefile
index 556ef35..d75adc5 100644
--- a/crypto/x509v3/Makefile
+++ b/crypto/x509v3/Makefile
@@ -10,7 +10,7 @@ CFLAG=-g
 MAKEFILE=	Makefile
 AR=		ar r
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README
 TEST=
diff --git a/engines/Makefile b/engines/Makefile
index e024205..605253c 100644
--- a/engines/Makefile
+++ b/engines/Makefile
@@ -19,7 +19,7 @@ RECURSIVE_MAKE=	[ -z "$(ENGDIRS)" ] || for i in $(ENGDIRS) ; do \
 PEX_LIBS=
 EX_LIBS=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile engines.com install.com engine_vector.mar
 TEST=
diff --git a/engines/ccgost/Makefile b/engines/ccgost/Makefile
index 64be962..43b5fd3 100644
--- a/engines/ccgost/Makefile
+++ b/engines/ccgost/Makefile
@@ -5,7 +5,7 @@ INCLUDES= -I../../include
 CFLAG=-g
 MAKEFILE= Makefile
 AR= ar r
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 LIB=$(TOP)/libcrypto.a
 
 LIBSRC= gost2001.c gost2001_keyx.c gost89.c gost94_keyx.c gost_ameth.c gost_asn1.c gost_crypt.c gost_ctl.c gost_eng.c gosthash.c gost_keywrap.c gost_md.c gost_params.c gost_pmeth.c gost_sign.c
diff --git a/ssl/Makefile b/ssl/Makefile
index 2b275fa..ede2edc 100644
--- a/ssl/Makefile
+++ b/ssl/Makefile
@@ -12,7 +12,7 @@ AR=		ar r
 # KRB5 stuff
 KRB5_INCLUDES=
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile README ssl-lib.com install.com
 TEST=ssltest.c
diff --git a/test/Makefile b/test/Makefile
index 3912f82..b3497fa 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -16,7 +16,7 @@ LIBKRB5=
 PEX_LIBS=
 EX_LIBS= #-lnsl -lsocket
 
-CFLAGS= $(INCLUDES) $(CFLAG)
+CFLAGS= $(EXTRA_CPPFLAGS) $(INCLUDES) $(CFLAG) $(EXTRA_CFLAGS)
 
 GENERAL=Makefile maketests.com \
 	tests.com testenc.com tx509.com trsa.com tcrl.com tsid.com treq.com \
-- 
1.7.3.2

