From 628ccc4f571c614581bc9d8efb4fb67c465f4ca4 Mon Sep 17 00:00:00 2001
From: David Schleef <ds@schleef.org>
Date: Mon, 15 Apr 2013 16:01:29 -0700
Subject: [PATCH 6/8] mpegtsdemux: check value before using as denominator

---
 gst/mpegtsdemux/mpegtspacketizer.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gst/mpegtsdemux/mpegtspacketizer.c b/gst/mpegtsdemux/mpegtspacketizer.c
index 83dee0e..5827a86 100644
--- a/gst/mpegtsdemux/mpegtspacketizer.c
+++ b/gst/mpegtsdemux/mpegtspacketizer.c
@@ -3655,6 +3655,9 @@ mpegts_packetizer_offset_to_ts (MpegTSPacketizer2 * packetizer, guint64 offset,
 
   pcrtable = get_pcr_table (packetizer, pid);
 
+  if (G_UNLIKELY (pcrtable->last_offset <= pcrtable->first_offset))
+    return GST_CLOCK_TIME_NONE;
+
   /* Convert byte difference into time difference */
   res = PCRTIME_TO_GSTTIME (gst_util_uint64_scale (offset - priv->refoffset,
           pcrtable->last_pcr - pcrtable->first_pcr,
-- 
1.8.1.4

