From f8769748e112b94a1f06036575bac80a4e8abb2a Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Mon, 19 Dec 2016 23:37:18 +0100
Subject: [PATCH] fix out-of-tree-build

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index f003bae..3d8f9cb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,7 +24,7 @@ sshfs.1: sshfs.1.in
 	$(AM_V_GEN)sed \
 	    -e 's,__IDMAP_DEFAULT__,$(IDMAP_DEFAULT),g' \
 	    -e 's,__UNMOUNT_COMMAND__,$(UNMOUNT_COMMAND),g' \
-	    <sshfs.1.in >sshfs.1.tmp || exit 1; \
+	    <$(srcdir)/sshfs.1.in >sshfs.1.tmp || exit 1; \
 	mv sshfs.1.tmp sshfs.1
 
 if SSH_NODELAY_SO
-- 
2.7.4

