diff --git a/configure.ac b/configure.ac
index 5a84725..b4c86b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,8 +50,21 @@ AC_ARG_ENABLE([tegra], AC_HELP_STRING([--enable-tegra]), [TEGRA=$enableval], [TE
 AM_CONDITIONAL([TEGRA], [test x$TEGRA = xyes])
 if test x$TEGRA = xyes; then
         AC_DEFINE([TEGRA], , [Enable tegra elements])
+        TEGRA_X11_INCLUDES=-I=/usr/include
+        TEGRA_X11_LIBS=-lX11
+        TEGRA_EGL_INCLUDES=-I=/usr/include
+        TEGRA_EGL_LIBS=-lEGL
+        TEGRA_OPENGL_INCLUDES=-I=/usr/include
+        TEGRA_OPENGL_LIBS=-lGLESv2
 fi
 
+AC_SUBST(TEGRA_X11_CFLAGS)
+AC_SUBST(TEGRA_X11_LIBS)
+AC_SUBST(TEGRA_EGL_CFLAGS)
+AC_SUBST(TEGRA_EGL_LIBS)
+AC_SUBST(TEGRA_OPENGL_CFLAGS)
+AC_SUBST(TEGRA_OPENGL_LIBS)
+
 dnl ** checks **
 
 dnl Check for GLib
diff --git a/omx/Makefile.am b/omx/Makefile.am
index 6f39632..974046e 100644
--- a/omx/Makefile.am
+++ b/omx/Makefile.am
@@ -63,7 +63,7 @@ libgstomx_la_CFLAGS = \
 	-I$(srcdir)/headers \
 	$(GST_CFLAGS) $(GST_BASE_CFLAGS) \
 	-I$(top_srcdir)/util \
-	-I$(TEGRA_X11_INCLUDES) -I$(TEGRA_EGL_INCLUDES) -I$(TEGRA_OPENGL_INCLUDES)
+	$(TEGRA_X11_CFLAGS) $(TEGRA_EGL_CFLAGS) $(TEGRA_OPENGL_CFLAGS)
 
 libgstomx_la_LIBADD = \
 	$(GST_LIBS) $(GST_BASE_LIBS) $(GST_INTERFACES_LIBS) \
diff --git a/omx/gstgl_videosink.c b/omx/gstgl_videosink.c
index 26df5fb..a197bd8 100644
--- a/omx/gstgl_videosink.c
+++ b/omx/gstgl_videosink.c
@@ -107,9 +107,8 @@ render (GstBaseSink *gst_base,
     memcpy(&eglbuf,GST_BUFFER_DATA(buf),sizeof(nvomx_eglbuffer));
     if (self->bvideodisable)
     {
-        EGLBoolean eglStatus;
 #ifdef EGL_NV_sync
-        eglStatus = eglFenceNV(eglbuf.hReadEglSync);
+        (void)eglFenceNV(eglbuf.hReadEglSync);
 #endif
         return GST_FLOW_OK;
     }
diff --git a/omx/gstomx_aacdec.c b/omx/gstomx_aacdec.c
index fd9767b..5fb44df 100644
--- a/omx/gstomx_aacdec.c
+++ b/omx/gstomx_aacdec.c
@@ -209,8 +209,7 @@ static GstFlowReturn gst_omx_aacdec_pad_chain (GstPad *pad, GstBuffer *buf)
     GstOmxBaseFilter *omx_base;
     GstOmxAacDec *omx_aacdec;
     GstFlowReturn result = GST_FLOW_ERROR;
-    int ID, object_type, index, protection_absent;
-    unsigned int frame_length;
+    int object_type, index, protection_absent;
     int skip_bytes = 0;
     GstBuffer *temp;
     omx_base = GST_OMX_BASE_FILTER (GST_PAD_PARENT (pad));
@@ -220,12 +219,9 @@ static GstFlowReturn gst_omx_aacdec_pad_chain (GstPad *pad, GstBuffer *buf)
         //Parse and remove the ADTS header
         if((buf->data[0]=0xff && (buf->data[1]&0xf6)==0xf0))
         {
-            ID = buf->data[1] & 0x08;
-               protection_absent = buf->data[1]&1;
+            protection_absent = buf->data[1]&1;
             object_type   = (buf->data[2] & 0xc0)>>6;    
             index = (buf->data[2] & 0x3c)>>2;
-            frame_length = ((buf->data[3]&3)<<11) | \
-                           buf->data[4]<<3 | (buf->data[5]&0xe0 >> 5) ;
             skip_bytes += 7;
             if(protection_absent == 0)
             {
diff --git a/omx/gstomx_camera.c b/omx/gstomx_camera.c
index ee814ef..00db53f 100644
--- a/omx/gstomx_camera.c
+++ b/omx/gstomx_camera.c
@@ -206,12 +206,6 @@ static gboolean
 setcaps (GstBaseSrc *gst_src,
          GstCaps *caps)
 {
-    GstOmxBaseSrc *self;
-
-    self = GST_OMX_BASE_SRC (gst_src);
-
-    //GST_INFO_OBJECT (self, "setcaps (src): %" GST_PTR_FORMAT, caps);
-
     g_return_val_if_fail (gst_caps_get_size (caps) == 1, FALSE);
 
     return TRUE;
@@ -220,12 +214,6 @@ setcaps (GstBaseSrc *gst_src,
 static void
 settings_changed_cb (GOmxCore *core)
 {
-    GstOmxBaseSrc *omx_base;
-
-    omx_base = core->object;
-
-
-    /** @todo properly set the capabilities */
 }
 
 static void
@@ -235,7 +223,6 @@ setup_ports (GstOmxBaseSrc *base_src)
     OMX_PARAM_PORTDEFINITIONTYPE param;
     OMX_PARAM_U32TYPE  oSensorIdParam;
     OMX_INDEXTYPE  eIndexSensorId;
-    OMX_ERRORTYPE  err;
     OMX_IMAGE_PARAM_FLASHCONTROLTYPE    oFlashControl;
     OMX_PARAM_SENSORMODETYPE     oSensorMode;
 
@@ -273,10 +260,10 @@ setup_ports (GstOmxBaseSrc *base_src)
 
     oSensorIdParam.nU32 = g_cameraParam.SensorId;
 
-    err = OMX_GetExtensionIndex(core->omx_handle, NVX_INDEX_PARAM_SENSORID,
+    (void)OMX_GetExtensionIndex(core->omx_handle, NVX_INDEX_PARAM_SENSORID,
                                    &eIndexSensorId);
 
-    err = OMX_SetParameter(core->omx_handle, eIndexSensorId, &oSensorIdParam);
+    (void)OMX_SetParameter(core->omx_handle, eIndexSensorId, &oSensorIdParam);
 
     memset (&oSensorMode, 0xDE, sizeof (oSensorMode));
     oSensorMode.nPortIndex = CAPTURE_PORT;
@@ -461,11 +448,10 @@ create (GstBaseSrc *gst_base,
     {
         OMX_CONFIG_BOOLEANTYPE     cc;
         OMX_INDEXTYPE eIndexPreviewEnable;
-        OMX_ERRORTYPE err;
   
  
         // start preview
-        err = OMX_GetExtensionIndex(core->omx_handle, NVX_INDEX_CONFIG_PREVIEWENABLE,
+        (void)OMX_GetExtensionIndex(core->omx_handle, NVX_INDEX_CONFIG_PREVIEWENABLE,
                                        &eIndexPreviewEnable);
         cc.bEnabled = OMX_TRUE;
         OMX_SetConfig(core->omx_handle, eIndexPreviewEnable, &cc);
@@ -524,9 +510,6 @@ type_instance_init (GTypeInstance *instance,
                     gpointer g_class)
 {
     GstOmxBaseSrc *omx_base;
-    GstOmxCamera  *gstcamera;
-
-    gstcamera = GST_OMX_CAMERA(instance);
 
     omx_base = GST_OMX_BASE_SRC (instance);
 
diff --git a/omx/gstomx_filereadersrc.c b/omx/gstomx_filereadersrc.c
index 744efa3..ec7b109 100644
--- a/omx/gstomx_filereadersrc.c
+++ b/omx/gstomx_filereadersrc.c
@@ -74,12 +74,6 @@ static gboolean
 setcaps (GstBaseSrc *gst_src,
          GstCaps *caps)
 {
-    GstOmxBaseSrc *self;
-
-    self = GST_OMX_BASE_SRC (gst_src);
-
-    //GST_INFO_OBJECT (self, "setcaps (src): %" GST_PTR_FORMAT, caps);
-
     g_return_val_if_fail (gst_caps_get_size (caps) == 1, FALSE);
 
     return TRUE;
@@ -88,13 +82,6 @@ setcaps (GstBaseSrc *gst_src,
 static void
 settings_changed_cb (GOmxCore *core)
 {
-    GstOmxBaseSrc *omx_base;
-
-    omx_base = core->object;
-
-
-printf("setting changed is done...\n");
-    /** @todo properly set the capabilities */
 }
 
 static void
diff --git a/omx/gstomx_overlaysink.c b/omx/gstomx_overlaysink.c
index a019206..6d5e0ce 100644
--- a/omx/gstomx_overlaysink.c
+++ b/omx/gstomx_overlaysink.c
@@ -126,11 +126,9 @@ setcaps (GstBaseSink *gst_sink,
          GstCaps *caps)
 {
     GstOmxBaseSink *omx_base;
-    GstOmxOverlaySink *self;
     GOmxCore *gomx;
     GOmxPort *port;
     omx_base = GST_OMX_BASE_SINK (gst_sink);
-    self = GST_OMX_OVERLAYSINK (gst_sink);
     gomx = (GOmxCore *) omx_base->gomx;
     GST_INFO_OBJECT (omx_base, "setcaps (sink): %" GST_PTR_FORMAT, caps);
 
diff --git a/omx/gstomx_videosink.c b/omx/gstomx_videosink.c
index 1b83da8..6d9f2ba 100644
--- a/omx/gstomx_videosink.c
+++ b/omx/gstomx_videosink.c
@@ -138,11 +138,9 @@ setcaps (GstBaseSink *gst_sink,
          GstCaps *caps)
 {
     GstOmxBaseSink *omx_base;
-    GstOmxVideoSink *self;
     GOmxCore *gomx;
     GOmxPort *port;
     omx_base = GST_OMX_BASE_SINK (gst_sink);
-    self = GST_OMX_VIDEOSINK (gst_sink);
     gomx = (GOmxCore *) omx_base->gomx;
     GST_INFO_OBJECT (omx_base, "setcaps (sink): %" GST_PTR_FORMAT, caps);
 
diff --git a/omx/gstomx_vip.c b/omx/gstomx_vip.c
index 815da60..b3ee108 100644
--- a/omx/gstomx_vip.c
+++ b/omx/gstomx_vip.c
@@ -120,11 +120,8 @@ static gboolean
 setcaps (GstBaseSrc *gst_src,
          GstCaps *caps)
 {
-    GstOmxBaseSrc *self;
     guint32 format = 0;
 
-    self = GST_OMX_BASE_SRC (gst_src);
-
     g_return_val_if_fail (gst_caps_get_size (caps) == 1, FALSE);
 
     {
@@ -175,12 +172,6 @@ setcaps (GstBaseSrc *gst_src,
 static void
 settings_changed_cb (GOmxCore *core)
 {
-    GstOmxBaseSrc *omx_base;
-
-    omx_base = core->object;
-
-
-    /** @todo properly set the capabilities */
 }
 
 static void
@@ -413,10 +404,6 @@ type_instance_init (GTypeInstance *instance,
                     gpointer g_class)
 {
     GstOmxBaseSrc *omx_base;
-    GstOmxVip  *gstvip;
-
-
-    gstvip = GST_OMX_VIP(instance);
 
     omx_base = GST_OMX_BASE_SRC (instance);
 
diff --git a/omx/gstomx_walvideosink.c b/omx/gstomx_walvideosink.c
index 213bc83..9d52f64 100644
--- a/omx/gstomx_walvideosink.c
+++ b/omx/gstomx_walvideosink.c
@@ -433,10 +433,8 @@ type_class_init (gpointer g_class,
                  gpointer class_data)
 {
     GObjectClass *gobject_class;
-    GstElementClass *gstelement_class;
 
     gobject_class = (GObjectClass *) g_class;
-    gstelement_class = (GstElementClass *) g_class;
    
     parent_class = g_type_class_ref (GST_TYPE_BIN);
  
@@ -652,9 +650,6 @@ type_instance_init (GTypeInstance *instance,
                     gpointer g_class)
 {
     GstOmxWalVideoSink *self;
-    GstElementClass *element_class;
-
-    element_class = GST_ELEMENT_CLASS (g_class);
 
     self = GST_OMX_WALVIDEOSINK (instance);
 
