diff --git a/libubacktrace/Makefile.in b/libubacktrace/Makefile.in
index f75f68b..3de5ed8 100644
--- a/libubacktrace/Makefile.in
+++ b/libubacktrace/Makefile.in
@@ -12,7 +12,7 @@ CFLAGS-libubacktrace := -DNOT_IN_libc -DIS_IN_libubacktrace $(SSP_ALL_CFLAGS)
 
 LDFLAGS-libubacktrace.so := $(LDFLAGS) $(top_builddir)lib/libdl-$(VERSION).so
 
-ifeq ($(CONFIG_ARM_EABI),y)
+ifeq ($(CONFIG_ARM_EABI)$(UCLIBC_HAS_BACKTRACE),yy)
 LIBGCC += $(shell $(CC) -print-file-name=libgcc_eh.a)
 endif
 
