diff --git a/configure.ac b/configure.ac
index 7a3ee3c..75071b0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -162,9 +162,18 @@ CC_CHECK_FLAGS_APPEND([with_cflags], [CFLAGS], [\
         -fno-strict-aliasing \
         -fvisibility=hidden \
         -ffunction-sections \
-        -fdata-sections \
-        -fstack-protector \
-        --param=ssp-buffer-size=4])
+        -fdata-sections])
+
+AC_ARG_ENABLE([ssp],
+        AS_HELP_STRING([--disable-ssp], [Disable stack protector support]),
+        [], [enable_ssp=yes])
+
+if test "x$enable_ssp" = "xyes"; then
+      CC_CHECK_FLAGS_APPEND([with_cflags], [CFLAGS], [\
+              -fstack-protector \
+              --param=ssp-buffer-size=4])
+fi
+
 AC_SUBST([OUR_CFLAGS], $with_cflags)
 
 CC_CHECK_FLAGS_APPEND([with_cppflags], [CPPFLAGS], [\
