diff --git a/Makefile.in b/Makefile.in
index b1d44c0..433f2d4 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -34,6 +34,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
+@ENABLE_TESTS_TRUE@am__append_1 = tests
 subdir = .
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/config.h.in \
@@ -98,7 +99,7 @@ AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=) \
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = $(SUBDIRS)
+DIST_SUBDIRS = libnotify docs tools tests
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -280,7 +281,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = libnotify docs tools tests
+SUBDIRS = libnotify docs tools $(am__append_1)
 pcdata_DATA = libnotify.pc
 pcdatadir = $(libdir)/pkgconfig
 EXTRA_DIST = \
diff --git a/aclocal.m4 b/aclocal.m4
index cf07a0a..9520927 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -251,7 +251,7 @@ $$1_PKG_ERRORS
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-_PKG_TEXT])
+_PKG_TEXT])[]dnl
         ])
 elif test $pkg_failed = untried; then
      	AC_MSG_RESULT([no])
@@ -262,7 +262,7 @@ path to pkg-config.
 
 _PKG_TEXT
 
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.])
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.])dnl
         ])
 else
 	$1[]_CFLAGS=$pkg_cv_[]$1[]_CFLAGS
diff --git a/configure b/configure
index 21f4040..b55aca8 100755
--- a/configure
+++ b/configure
@@ -646,6 +646,8 @@ INTROSPECTION_GENERATE
 INTROSPECTION_COMPILER
 INTROSPECTION_SCANNER
 GLIB_GENMARSHAL
+ENABLE_TESTS_FALSE
+ENABLE_TESTS_TRUE
 TESTS_LIBS
 TESTS_CFLAGS
 pkg_modules
@@ -782,6 +784,7 @@ enable_fast_install
 with_gnu_ld
 with_sysroot
 enable_libtool_lock
+enable_tests
 enable_introspection
 with_html_dir
 enable_gtk_doc
@@ -799,7 +802,6 @@ LDFLAGS
 LIBS
 CPPFLAGS
 CPP
-CPPFLAGS
 PKG_CONFIG
 PKG_CONFIG_PATH
 PKG_CONFIG_LIBDIR
@@ -1438,6 +1440,7 @@ Optional Features:
   --enable-fast-install[=PKGS]
                           optimize for fast installation [default=yes]
   --disable-libtool-lock  avoid locking (might break parallel builds)
+  --disable-tests         build tests (requires gtk+-3.0)
   --enable-introspection=[no/auto/yes]
                           Enable introspection for this build
   --enable-gtk-doc        use gtk-doc to build documentation [[default=no]]
@@ -11634,7 +11637,6 @@ installed software in a non-standard prefix.
 Alternatively, you may set the environment variables PACKAGE_CFLAGS
 and PACKAGE_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -11650,7 +11652,6 @@ See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	PACKAGE_CFLAGS=$pkg_cv_PACKAGE_CFLAGS
 	PACKAGE_LIBS=$pkg_cv_PACKAGE_LIBS
@@ -11662,7 +11663,15 @@ fi
 
 
 
-tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
+# Check whether --enable-tests was given.
+if test "${enable_tests+set}" = set; then :
+  enableval=$enable_tests; enable_tests=$enableval
+else
+  enable_tests=yes
+fi
+
+if test x$enable_tests = xyes; then
+	tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for TESTS" >&5
@@ -11730,7 +11739,6 @@ installed software in a non-standard prefix.
 Alternatively, you may set the environment variables TESTS_CFLAGS
 and TESTS_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -11746,7 +11754,6 @@ See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	TESTS_CFLAGS=$pkg_cv_TESTS_CFLAGS
 	TESTS_LIBS=$pkg_cv_TESTS_LIBS
@@ -11756,6 +11763,15 @@ $as_echo "yes" >&6; }
 fi
 
 
+fi
+ if test x$enable_tests = xyes; then
+  ENABLE_TESTS_TRUE=
+  ENABLE_TESTS_FALSE='#'
+else
+  ENABLE_TESTS_TRUE='#'
+  ENABLE_TESTS_FALSE=
+fi
+
 
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 
@@ -12358,6 +12374,10 @@ if test -z "${am__fastdepCC_TRUE}" && test -z "${am__fastdepCC_FALSE}"; then
   as_fn_error $? "conditional \"am__fastdepCC\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
+if test -z "${ENABLE_TESTS_TRUE}" && test -z "${ENABLE_TESTS_FALSE}"; then
+  as_fn_error $? "conditional \"ENABLE_TESTS\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 if test -z "${HAVE_INTROSPECTION_TRUE}" && test -z "${HAVE_INTROSPECTION_FALSE}"; then
   as_fn_error $? "conditional \"HAVE_INTROSPECTION\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
