diff --git a/configure.ac b/configure.ac
index 1901bcf..e53eb27 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,6 +22,7 @@ AS_NANO(GST_GIT="no", GST_GIT="yes")
 
 dnl can autoconf find the source ?
 AC_CONFIG_SRCDIR([gst/audiotestsrc/gstaudiotestsrc.c])
+AC_CONFIG_MACRO_DIR([m4])
 
 dnl define the output header for config
 AM_CONFIG_HEADER([config.h])
diff --git a/gst-libs/gst/audio/Makefile.am b/gst-libs/gst/audio/Makefile.am
index 6e89c47..84b82b6 100644
--- a/gst-libs/gst/audio/Makefile.am
+++ b/gst-libs/gst/audio/Makefile.am
@@ -54,8 +54,8 @@ nodist_libgstaudio_@GST_MAJORMINOR@include_HEADERS = \
 
 libgstaudio_@GST_MAJORMINOR@_la_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GST_CFLAGS)
 libgstaudio_@GST_MAJORMINOR@_la_LIBADD = \
-  $(top_builddir)/gst-libs/gst/pbutils/libgstpbutils-@GST_MAJORMINOR@.la \
-  $(top_builddir)/gst-libs/gst/interfaces/libgstinterfaces-@GST_MAJORMINOR@.la \
+  ../pbutils/libgstpbutils-@GST_MAJORMINOR@.la \
+  ../interfaces/libgstinterfaces-@GST_MAJORMINOR@.la \
   $(GST_BASE_LIBS) $(GST_LIBS)
 libgstaudio_@GST_MAJORMINOR@_la_LDFLAGS = $(GST_LIB_LDFLAGS) $(GST_ALL_LDFLAGS) $(GST_LT_LDFLAGS)
 
