diff --git a/gcc/Makefile.in b/gcc/Makefile.in
index b9e1858..f43a60a 100644
--- a/gcc/Makefile.in
+++ b/gcc/Makefile.in
@@ -377,7 +377,7 @@ PARTITION_H = $(srcdir)/../include/partition.h
 MD5_H	    = $(srcdir)/../include/md5.h
 
 # Default native SYSTEM_HEADER_DIR, to be overridden by targets.
-NATIVE_SYSTEM_HEADER_DIR = /usr/include
+NATIVE_SYSTEM_HEADER_DIR = $(prefix)/$(target_noncanonical)/include
 # Default cross SYSTEM_HEADER_DIR, to be overridden by targets.
 CROSS_SYSTEM_HEADER_DIR = @CROSS_SYSTEM_HEADER_DIR@
 
