--- a/src/configure.in	2007-08-24 11:33:30.000000000 +0200
+++ b/src/configure.in	2007-08-24 11:33:43.000000000 +0200
@@ -47,13 +47,6 @@
   fi
 fi
 
-dnl If configure thinks we are cross compiling, there is probably something
-dnl wrong with the CC or CFLAGS settings, give an understandable error message
-if test "$cross_compiling" = yes; then
-  AC_MSG_ERROR([cannot compile a simple program, check CC and CFLAGS
-  (cross compiling doesn't work)])
-fi
-
 dnl gcc-cpp has the wonderful -MM option to produce nicer dependencies.
 dnl But gcc 3.1 changed the meaning!  See near the end.
 test "$GCC" = yes && CPP_MM=M; AC_SUBST(CPP_MM)
--- a/src/auto/configure	2007-08-24 11:34:26.000000000 +0200
+++ b/src/auto/configure	2007-08-24 11:34:41.000000000 +0200
@@ -2796,14 +2796,6 @@
   fi
 fi
 
-if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot compile a simple program, check CC and CFLAGS
-  (cross compiling doesn't work)" >&5
-echo "$as_me: error: cannot compile a simple program, check CC and CFLAGS
-  (cross compiling doesn't work)" >&2;}
-   { (exit 1); exit 1; }; }
-fi
-
 test "$GCC" = yes && CPP_MM=M;
 
 if test -f ./toolcheck; then
