--- a/Rules.mak	2007-06-09 20:59:22.000000000 +0200
+++ b/Rules.mak	2007-06-09 21:00:08.000000000 +0200
@@ -209,8 +209,7 @@
 	CPU_CFLAGS-$(CONFIG_ARM1176JZF_S)+=-mtune=arm1176jzf-s -march=armv6
 	CPU_CFLAGS-$(CONFIG_ARM_SA110)+=-mtune=strongarm110 -march=armv4
 	CPU_CFLAGS-$(CONFIG_ARM_SA1100)+=-mtune=strongarm1100 -march=armv4
-	CPU_CFLAGS-$(CONFIG_ARM_XSCALE)+=$(call check_gcc,-mtune=xscale,-mtune=strongarm110)
-	CPU_CFLAGS-$(CONFIG_ARM_XSCALE)+=-march=armv5te -Wa,-mcpu=xscale
+	CPU_CFLAGS-$(CONFIG_ARM_XSCALE)+=-march=armv5te -mtune=xscale -Wa,-mcpu=xscale
  	CPU_CFLAGS-$(CONFIG_ARM_IWMMXT)+=-march=iwmmxt -Wa,-mcpu=iwmmxt -mabi=iwmmxt
 endif
 
