diff --git a/configure b/configure
index 26e239d..3ab6612 100755
--- a/configure
+++ b/configure
@@ -11654,79 +11654,6 @@ fi
 
 
 
-  { echo "$as_me:$LINENO: checking if compiler supports gcc-style inline assembly" >&5
-echo $ECHO_N "checking if compiler supports gcc-style inline assembly... $ECHO_C" >&6; }
-
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-int
-main ()
-{
-
-#ifndef __GNUC__
-#error Not GCC
-#endif
-#ifdef __GNUC_MINOR__
-#if (__GNUC__ * 1000 + __GNUC_MINOR__) < 3004
-#error GCC before 3.4 has critical bugs compiling inline assembly
-#endif
-#endif
-__asm__ (""::)
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext
-if { (ac_try="$ac_compile"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
-  (eval "$ac_compile") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest.$ac_objext; then
-  flag_ok=yes
-else
-  echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	flag_ok=no
-fi
-
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-
-  if test "X$flag_ok" = Xyes ; then
-    HAVE_GCC_ASM=yes
-    true
-  else
-    HAVE_GCC_ASM=no
-    true
-  fi
-  { echo "$as_me:$LINENO: result: $flag_ok" >&5
-echo "${ECHO_T}$flag_ok" >&6; }
-
-if test x$HAVE_GCC_ASM = xyes ; then
-
-cat >>confdefs.h <<\_ACEOF
-#define HAVE_GCC_ASM 1
-_ACEOF
-
-fi
-
-
 if test "x$HAVE_GCC_ASM" = "xyes"; then
   HAVE_GCC_ASM_TRUE=
   HAVE_GCC_ASM_FALSE='#'
@@ -17456,73 +17383,6 @@ echo "${ECHO_T}$flag_ok" >&6; }
 
 fi
 
-if test x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes ; then
-
-  { echo "$as_me:$LINENO: checking to see if compiler understands \"-Wa,-mfpu=vfp\"" >&5
-echo $ECHO_N "checking to see if compiler understands \"-Wa,-mfpu=vfp\"... $ECHO_C" >&6; }
-
-  save_CFLAGS="$CFLAGS"
-  CFLAGS="$CFLAGS "-Wa,-mfpu=vfp""
-
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext
-if { (ac_try="$ac_compile"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval "echo \"\$as_me:$LINENO: $ac_try_echo\"") >&5
-  (eval "$ac_compile") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest.$ac_objext; then
-  flag_ok=yes
-else
-  echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	flag_ok=no
-fi
-
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-  CFLAGS="$save_CFLAGS"
-
-  if test "X$flag_ok" = Xyes ; then
-    VFP_CFLAGS="$VFP_CFLAGS -mfpu=vfp"
-    true
-  else
-    true
-    true
-  fi
-  { echo "$as_me:$LINENO: result: $flag_ok" >&5
-echo "${ECHO_T}$flag_ok" >&6; }
-
-  #AS_COMPILER_FLAG(["-Wa,-mfloat-abi=softfp"],
-  #  [VFP_CFLAGS="$VFP_CFLAGS -mfloat-abi=softfp"],
-  #  true)
-fi
-
-
 #AS_COMPILER_FLAG(["-fasm-blocks"], HAVE_ASM_BLOCKS=yes, HAVE_ASM_BLOCKS=no)
 HAVE_ASM_BLOCKS=no
 if test "x$HAVE_ASM_BLOCKS" = "xyes"
