--- a/Makefile.in	2007-07-19 12:37:02.000000000 +0200
+++ b/Makefile.in	2007-07-19 12:37:42.000000000 +0200
@@ -117,7 +117,6 @@
 libpcrecpp_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
 	$(CXXFLAGS) $(libpcrecpp_la_LDFLAGS) $(LDFLAGS) -o $@
-@WITH_PCRE_CPP_TRUE@am_libpcrecpp_la_rpath = -rpath $(libdir)
 libpcreposix_la_DEPENDENCIES = libpcre.la
 am_libpcreposix_la_OBJECTS = pcreposix.lo
 libpcreposix_la_OBJECTS = $(am_libpcreposix_la_OBJECTS)
@@ -678,11 +677,11 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libpcre.la: $(libpcre_la_OBJECTS) $(libpcre_la_DEPENDENCIES)
-	$(libpcre_la_LINK) -rpath $(libdir) $(libpcre_la_OBJECTS) $(libpcre_la_LIBADD) $(LIBS)
+	$(libpcre_la_LINK) $(libpcre_la_OBJECTS) $(libpcre_la_LIBADD) $(LIBS)
 libpcrecpp.la: $(libpcrecpp_la_OBJECTS) $(libpcrecpp_la_DEPENDENCIES)
-	$(libpcrecpp_la_LINK) $(am_libpcrecpp_la_rpath) $(libpcrecpp_la_OBJECTS) $(libpcrecpp_la_LIBADD) $(LIBS)
+	$(libpcrecpp_la_LINK) $(libpcrecpp_la_OBJECTS) $(libpcrecpp_la_LIBADD) $(LIBS)
 libpcreposix.la: $(libpcreposix_la_OBJECTS) $(libpcreposix_la_DEPENDENCIES)
-	$(libpcreposix_la_LINK) -rpath $(libdir) $(libpcreposix_la_OBJECTS) $(libpcreposix_la_LIBADD) $(LIBS)
+	$(libpcreposix_la_LINK) $(libpcreposix_la_OBJECTS) $(libpcreposix_la_LIBADD) $(LIBS)
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(MKDIR_P) "$(DESTDIR)$(bindir)"
