diff --git a/Makerules b/Makerules
index 7153836..e9c3564 100644
--- a/Makerules
+++ b/Makerules
@@ -243,7 +243,7 @@ cmd_compile.mi= $(cmd_compile.m:-c=-E -dD) $(UCLIBC_EXTRA_CPPFLAGS)
 cmd_compile-m = $(CC) $^ -c -o $@ $(CFLAGS) $(CFLAGS-$(suffix $@)) $(CFLAGS-$(notdir $(@D))) $(CFLAGS-$(notdir $@)) $(sort $(foreach d,$(^:$(top_srcdir)=),$(collect_multi_flags)))
 cmd_strip     = $(STRIPTOOL) $(STRIP_FLAGS) $^
 cmd_t_strip   = $(STRIPTOOL) $(STRIP_FLAGS) $@
-cmd_ar        = $(AR) $(ARFLAGS) $@ $^
+cmd_ar        = (cd $(top_builddir) && $(AR) $(ARFLAGS) $@ $(subst $(top_builddir),,$^))
 
 define do_ln
 	@$(disp_ln)
