diff --git a/utils/Makefile b/utils/Makefile
index bdaea9f..bdd9519 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -6,8 +6,8 @@
 #
 
 top_srcdir=../
-top_builddir=../
-include $(top_builddir)Rules.mak
+top_builddir=$(if $(O),$(O),..)/
+include $(top_srcdir)Rules.mak
 all: utils
 include Makefile.in
 include $(top_srcdir)Makerules
diff --git a/utils/Makefile.in b/utils/Makefile.in
index 65364d7..217f01d 100644
--- a/utils/Makefile.in
+++ b/utils/Makefile.in
@@ -99,7 +99,11 @@ utils: $(utils_OBJ) $(utils_LOCALE_OBJ)
 
 # NOTE: We build the utils AFTER we have a uClibc-targeted toolchain.
 
-$(utils_OBJ): $(utils_OUT)/% : $(utils_DIR)/%.c | $(libc)
+subdirs: $(addprefix $(top_builddir),$(subdirs))
+$(addprefix $(top_builddir),$(subdirs)):
+	$(do_mkdir)
+
+$(utils_OBJ): $(utils_OUT)/% : $(utils_DIR)/%.c | $(libc) subdirs
 	$(compile.u)
 
 $(utils_OUT)/locale: $(top_srcdir)extra/locale/programs/locale.c | $(libc)
