diff --git a/Makefile.am b/Makefile.am
index 0c8fb69..81864eb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,38 +1,12 @@
 include $(top_srcdir)/config/include-Makefile.am
 
 SUBDIRS = \
-	include \
-	disk-utils \
-	fdisk \
-	getopt \
-	lib \
-	shlibs \
-	login-utils \
-	misc-utils \
-	po \
-	schedutils \
-	sys-utils \
-	text-utils \
-	tests
+	shlibs
 
 
 RCS_FIND_IGNORE := \( -name SCCS -o -name BitKeeper -o -name .svn -o \
 		-name CVS -o -name .pc -o -name .hg -o -name .git \) -prune -o
 
-if LINUX
-SUBDIRS += \
-	hwclock \
-	partx
-endif
-
-if BUILD_MOUNT
-SUBDIRS += mount
-endif
-
-if BUILD_FSCK
-SUBDIRS += fsck
-endif
-
 ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = \
diff --git a/Makefile.in b/Makefile.in
index 046a3a2..a482e60 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -41,12 +41,7 @@ DIST_COMMON = README $(am__configure_deps) $(dist_noinst_DATA) \
 	TODO config/config.guess config/config.rpath config/config.sub \
 	config/depcomp config/install-sh config/ltmain.sh \
 	config/missing config/mkinstalldirs config/texinfo.tex
-@LINUX_TRUE@am__append_1 = \
-@LINUX_TRUE@	hwclock \
-@LINUX_TRUE@	partx
 
-@BUILD_MOUNT_TRUE@am__append_2 = mount
-@BUILD_FSCK_TRUE@am__append_3 = fsck
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/m4/gettext.m4 \
@@ -288,9 +283,7 @@ ul_libblkid_la = $(top_builddir)/shlibs/blkid/src/libblkid.la
 ul_libuuid_srcdir = $(top_srcdir)/shlibs/uuid/src
 ul_libuuid_builddir = $(top_builddir)/shlibs/uuid/src
 ul_libuuid_la = $(top_builddir)/shlibs/uuid/src/libuuid.la
-SUBDIRS = include disk-utils fdisk getopt lib shlibs login-utils \
-	misc-utils po schedutils sys-utils text-utils tests \
-	$(am__append_1) $(am__append_2) $(am__append_3)
+SUBDIRS = shlibs
 RCS_FIND_IGNORE := \( -name SCCS -o -name BitKeeper -o -name .svn -o \
 		-name CVS -o -name .pc -o -name .hg -o -name .git \) -prune -o
 
