diff --git a/ltmain.sh b/ltmain.sh
index b4583e7..e6a09fd 100755
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -1283,6 +1283,14 @@ EOF
 	  prev=
 	  continue
 	  ;;
+	sysroot)
+	  lt_sysroot="$arg"
+	  compiler_flags="$compiler_flags --sysroot $arg"
+	  compile_command="$compile_command --sysroot $arg"
+	  finalize_command="$finalize_command --sysroot $arg"
+	  prev=
+	  continue
+	  ;;
 	precious_regex)
 	  precious_files_regex="$arg"
 	  prev=
@@ -1538,6 +1546,11 @@ EOF
 	continue
 	;;
 
+      --sysroot)
+	prev=sysroot
+	continue
+	;;
+
       # The native IRIX linker understands -LANG:*, -LIST:* and -LNO:*
       # so, if we see these flags be careful not to treat them like -L
       -L[A-Z][A-Z]*:*)
@@ -2152,7 +2165,7 @@ EOF
 	  for searchdir in $newlib_search_path $lib_search_path $sys_lib_search_path $shlib_search_path; do
 	    for search_ext in .la $std_shrext .so .a; do
 	      # Search the libtool library
-	      lib="$searchdir/lib${name}${search_ext}"
+	      lib="$lt_sysroot$searchdir/lib${name}${search_ext}"
 	      if test -f "$lib"; then
 		if test "$search_ext" = ".la"; then
 		  found=yes
@@ -5758,7 +5771,7 @@ dlopen='$dlfiles'
 dlpreopen='$dlprefiles'
 
 # Directory that this library needs to be installed in:
-libdir='$install_libdir'"
+libdir='$lt_sysroot$install_libdir'"
 	  if test "$installed" = no && test "$need_relink" = yes; then
 	    $echo >> $output "\
 relink_command=\"$relink_command\""
