diff --git a/Makefile.in b/Makefile.in
index 9868303..279dedd 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -68,16 +68,21 @@ host_triplet = @host@
 @TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_TRUE@@TARGET_GENTOO_FALSE@	-DKBD_SETFONT=\"/usr/bin/setfont\" \
 @TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_TRUE@@TARGET_GENTOO_FALSE@	-DDEFAULT_FONT=\"LatArCyrHeb-16\"
 
-@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@am__append_4 = \
-@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@	-DKBD_LOADKEYS=\"/bin/loadkeys\" \
-@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@	-DKBD_SETFONT=\"/bin/setfont\" \
-@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@	-DDEFAULT_FONT=\"latarcyrheb-sun16\"
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_TRUE@am__append_4 = \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_TRUE@	-DKBD_LOADKEYS=\"/usr/bin/loadkeys\" \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_TRUE@	-DKBD_SETFONT=\"/usr/bin/setfont\" \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_TRUE@	-DDEFAULT_FONT=\"default8x16\"
+
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_FALSE@am__append_5 = \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_FALSE@	-DKBD_LOADKEYS=\"/bin/loadkeys\" \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_FALSE@	-DKBD_SETFONT=\"/bin/setfont\" \
+@TARGET_ARCH_FALSE@@TARGET_FRUGALWARE_FALSE@@TARGET_GENTOO_FALSE@@TARGET_PBS_FALSE@	-DDEFAULT_FONT=\"latarcyrheb-sun16\"
 
 rootbin_PROGRAMS = systemd$(EXEEXT) systemctl$(EXEEXT) \
 	systemd-notify$(EXEEXT) systemd-ask-password$(EXEEXT) \
 	systemd-tty-ask-password-agent$(EXEEXT)
 bin_PROGRAMS = systemd-cgls$(EXEEXT) $(am__EXEEXT_1)
-@HAVE_GTK_TRUE@am__append_5 = \
+@HAVE_GTK_TRUE@am__append_6 = \
 @HAVE_GTK_TRUE@	systemadm \
 @HAVE_GTK_TRUE@	systemd-gnome-ask-password-agent
 
@@ -94,7 +99,7 @@ rootlibexec_PROGRAMS = systemd-logger$(EXEEXT) \
 	systemd-quotacheck$(EXEEXT) systemd-timestamp$(EXEEXT) \
 	systemd-ac-power$(EXEEXT) systemd-sysctl$(EXEEXT) \
 	$(am__EXEEXT_2)
-@HAVE_LIBCRYPTSETUP_TRUE@am__append_6 = \
+@HAVE_LIBCRYPTSETUP_TRUE@am__append_7 = \
 @HAVE_LIBCRYPTSETUP_TRUE@	systemd-cryptsetup
 
 @HAVE_LIBCRYPTSETUP_TRUE@systemgenerator_PROGRAMS = systemd-cryptsetup-generator$(EXEEXT)
@@ -102,7 +107,7 @@ noinst_PROGRAMS = test-engine$(EXEEXT) test-job-type$(EXEEXT) \
 	test-ns$(EXEEXT) test-loopback$(EXEEXT) test-hostname$(EXEEXT) \
 	test-daemon$(EXEEXT) test-cgroup$(EXEEXT) \
 	test-env-replace$(EXEEXT) test-strv$(EXEEXT)
-@TARGET_FEDORA_TRUE@am__append_7 = \
+@TARGET_FEDORA_TRUE@am__append_8 = \
 @TARGET_FEDORA_TRUE@	units/fedora/prefdm.service \
 @TARGET_FEDORA_TRUE@	units/fedora/rc-local.service \
 @TARGET_FEDORA_TRUE@	units/fedora/halt-local.service \
@@ -114,13 +119,13 @@ noinst_PROGRAMS = test-engine$(EXEEXT) test-job-type$(EXEEXT) \
 @TARGET_FEDORA_TRUE@	units/fedora/plymouth-poweroff.service \
 @TARGET_FEDORA_TRUE@	units/fedora/plymouth-halt.service
 
-@LIBNOTIFY07_TRUE@am__append_8 = \
+@LIBNOTIFY07_TRUE@am__append_9 = \
 @LIBNOTIFY07_TRUE@	-D LIBNOTIFY07
 
-@HAVE_VALAC_TRUE@am__append_9 = \
+@HAVE_VALAC_TRUE@am__append_10 = \
 @HAVE_VALAC_TRUE@	${systemadm_SOURCES:.vala=.c}
 
-@HAVE_XSLTPROC_TRUE@am__append_10 = \
+@HAVE_XSLTPROC_TRUE@am__append_11 = \
 @HAVE_XSLTPROC_TRUE@	$(dist_man_MANS) \
 @HAVE_XSLTPROC_TRUE@	${nodist_man_MANS:=.in} \
 @HAVE_XSLTPROC_TRUE@	${XML_FILES:.xml=.html} \
@@ -955,7 +960,7 @@ AM_CPPFLAGS = -include $(top_builddir)/config.h \
 	-DSYSTEM_GENERATOR_PATH=\"$(systemgeneratordir)\" \
 	-DUSER_GENERATOR_PATH=\"$(usergeneratordir)\" -I \
 	$(top_srcdir)/src $(am__append_1) $(am__append_2) \
-	$(am__append_3) $(am__append_4)
+	$(am__append_3) $(am__append_4) $(am__append_5)
 @HAVE_PAM_TRUE@pamlib_LTLIBRARIES = \
 @HAVE_PAM_TRUE@	pam_systemd.la
 
@@ -1019,7 +1024,7 @@ dist_systemunit_DATA = units/emergency.service units/emergency.target \
 	units/systemd-tmpfiles-clean.timer units/quotaon.service \
 	units/systemd-ask-password-wall.path \
 	units/systemd-ask-password-plymouth.path \
-	units/systemd-ask-password-console.path $(am__append_7)
+	units/systemd-ask-password-console.path $(am__append_8)
 nodist_systemunit_DATA = \
 	units/getty@.service \
 	units/serial-getty@.service \
@@ -1684,7 +1689,7 @@ systemd_gnome_ask_password_agent_CFLAGS = \
 
 systemd_gnome_ask_password_agent_VALAFLAGS = --pkg=posix \
 	--pkg=gtk+-2.0 --pkg=linux --pkg=gio-unix-2.0 --pkg=libnotify \
-	$(am__append_8) -g
+	$(am__append_9) -g
 systemd_gnome_ask_password_agent_LDADD = \
 	$(LIBNOTIFY_LIBS) \
 	$(GTK_LIBS)
@@ -1742,7 +1747,7 @@ M4_PROCESS_USER = \
 
 CLEANFILES = $(nodist_systemunit_DATA) $(nodist_userunit_DATA) \
 	$(nodist_man_MANS) ${XML_IN_FILES:.xml.in=.html} \
-	$(pkgconfigdata_DATA) $(am__append_9) $(am__append_10) \
+	$(pkgconfigdata_DATA) $(am__append_10) $(am__append_11) \
 	$(dbusinterface_DATA)
 @HAVE_XSLTPROC_TRUE@XSLTPROC_FLAGS = \
 @HAVE_XSLTPROC_TRUE@	--nonet \
diff --git a/aclocal.m4 b/aclocal.m4
index 09b2c53..c643a4f 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -156,7 +156,7 @@ $$1_PKG_ERRORS
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-_PKG_TEXT])
+_PKG_TEXT])[]dnl
         ])
 elif test $pkg_failed = untried; then
      	AC_MSG_RESULT([no])
@@ -167,7 +167,7 @@ path to pkg-config.
 
 _PKG_TEXT
 
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.])
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.])dnl
         ])
 else
 	$1[]_CFLAGS=$pkg_cv_[]$1[]_CFLAGS
diff --git a/config.h.in b/config.h.in
index 9cd3faf..baa6604 100644
--- a/config.h.in
+++ b/config.h.in
@@ -121,6 +121,9 @@
 /* Target is Gentoo */
 #undef TARGET_GENTOO
 
+/* Target is PBS */
+#undef TARGET_PBS
+
 /* Target is Slackware */
 #undef TARGET_SLACKWARE
 
diff --git a/configure b/configure
index 92c9bc0..dd8ca8e 100755
--- a/configure
+++ b/configure
@@ -683,6 +683,8 @@ dbusinterfacedir
 dbussystemservicedir
 dbussessionservicedir
 dbuspolicydir
+TARGET_PBS_FALSE
+TARGET_PBS_TRUE
 TARGET_FRUGALWARE_FALSE
 TARGET_FRUGALWARE_TRUE
 TARGET_SLACKWARE_FALSE
@@ -893,7 +895,6 @@ LDFLAGS
 LIBS
 CPPFLAGS
 CPP
-CPPFLAGS
 PKG_CONFIG
 PKG_CONFIG_PATH
 PKG_CONFIG_LIBDIR
@@ -1552,8 +1553,8 @@ Optional Packages:
   --with-sysroot=DIR Search for dependent libraries within DIR
                         (or the compiler's sysroot if not specified).
   --with-distro=DISTRO    Specify the distribution to target: One of fedora,
-                          suse, debian, ubuntu, arch, gentoo, slackware or
-                          other
+                          suse, debian, ubuntu, arch, gentoo, slackware, pbs
+                          or other
   --with-sysvinit-path=PATH
                           Specify the path to where the SysV init scripts are
                           located [default=based on distro]
@@ -12705,7 +12706,6 @@ installed software in a non-standard prefix.
 Alternatively, you may set the environment variables UDEV_CFLAGS
 and UDEV_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -12721,7 +12721,6 @@ See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	UDEV_CFLAGS=$pkg_cv_UDEV_CFLAGS
 	UDEV_LIBS=$pkg_cv_UDEV_LIBS
@@ -12799,7 +12798,6 @@ installed software in a non-standard prefix.
 Alternatively, you may set the environment variables DBUS_CFLAGS
 and DBUS_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -12815,7 +12813,6 @@ See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	DBUS_CFLAGS=$pkg_cv_DBUS_CFLAGS
 	DBUS_LIBS=$pkg_cv_DBUS_LIBS
@@ -13411,7 +13408,6 @@ installed software in a non-standard prefix.
 Alternatively, you may set the environment variables LIBNOTIFY_CFLAGS
 and LIBNOTIFY_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details." "$LINENO" 5
-
 elif test $pkg_failed = untried; then
      	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
@@ -13427,7 +13423,6 @@ See the pkg-config man page for more details.
 
 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
 See \`config.log' for more details" "$LINENO" 5; }
-
 else
 	LIBNOTIFY_CFLAGS=$pkg_cv_LIBNOTIFY_CFLAGS
 	LIBNOTIFY_LIBS=$pkg_cv_LIBNOTIFY_LIBS
@@ -13869,6 +13864,13 @@ $as_echo "#define TARGET_FRUGALWARE /**/" >>confdefs.h
 
                 M4_DISTRO_FLAG=-DTARGET_FRUGALWARE=1
                 ;;
+        pbs)
+                SYSTEM_SYSVRCND_PATH=/etc
+
+$as_echo "#define TARGET_PBS /**/" >>confdefs.h
+
+                M4_DISTRO_FLAG=-DTARGET_PBS=1
+                ;;
         other)
                 if test "x$with_syslog_service" = "x"; then :
   as_fn_error $? "With --distro=other, you must pass --with-syslog-service= to configure" "$LINENO" 5
@@ -13999,6 +14001,14 @@ else
   TARGET_FRUGALWARE_FALSE=
 fi
 
+ if test x"$with_distro" = xpbs; then
+  TARGET_PBS_TRUE=
+  TARGET_PBS_FALSE='#'
+else
+  TARGET_PBS_TRUE='#'
+  TARGET_PBS_FALSE=
+fi
+
 
 
 cat >>confdefs.h <<_ACEOF
@@ -14275,6 +14285,10 @@ if test -z "${TARGET_FRUGALWARE_TRUE}" && test -z "${TARGET_FRUGALWARE_FALSE}";
   as_fn_error $? "conditional \"TARGET_FRUGALWARE\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
+if test -z "${TARGET_PBS_TRUE}" && test -z "${TARGET_PBS_FALSE}"; then
+  as_fn_error $? "conditional \"TARGET_PBS\" was never defined.
+Usually this means the macro was only invoked conditionally." "$LINENO" 5
+fi
 
 : "${CONFIG_STATUS=./config.status}"
 ac_write_fail=0
