diff --git a/src/readahead/readahead-analyze.c b/src/readahead/readahead-analyze.c
index 9a929c0..b9a0ac3 100644
--- a/src/readahead/readahead-analyze.c
+++ b/src/readahead/readahead-analyze.c
@@ -135,7 +135,7 @@ int main_analyze(const char *pack_path) {
         printf("\nHOST:    %s"
                "TYPE:    %c\n"
                "MISSING: %d\n"
-               "TOTAL:   %llu\n",
+               "TOTAL:   %" PRId64 "\n",
                line,
                a,
                missing,
