diff --git a/src/journal/journald.c b/src/journal/journald.c
index baad3ab..1899ad6 100644
--- a/src/journal/journald.c
+++ b/src/journal/journald.c
@@ -2144,7 +2144,7 @@ static int process_event(Server *s, struct epoll_event *ev) {
                                 uint8_t buf[CMSG_SPACE(sizeof(struct ucred)) +
                                             CMSG_SPACE(sizeof(struct timeval)) +
                                             CMSG_SPACE(sizeof(int)) +
-                                            CMSG_SPACE(PAGE_SIZE)]; /* selinux label */
+                                            CMSG_SPACE(PATH_MAX)]; /* selinux label */
                         } control;
                         ssize_t n;
                         int v;
