From 8da3ea0db90c9f95d703dec0c9a24ad315f581f9 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Wed, 15 Feb 2012 08:05:13 +0100
Subject: [PATCH] Add PBS distribution support.

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 Makefile.am  |    7 +++++++
 configure.ac |    8 +++++++-
 2 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 9762da1..ea97d19 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -148,6 +148,12 @@ AM_CPPFLAGS += \
 	-DKBD_SETFONT=\"/bin/setfont\" \
 	-DDEFAULT_FONT=\"LatArCyrHeb-16\"
 else
+if TARGET_PBS
+AM_CPPFLAGS += \
+	-DKBD_LOADKEYS=\"/usr/bin/loadkeys\" \
+	-DKBD_SETFONT=\"/usr/bin/setfont\" \
+	-DDEFAULT_FONT=\"default8x16\"
+else
 AM_CPPFLAGS += \
 	-DKBD_LOADKEYS=\"/bin/loadkeys\" \
 	-DKBD_SETFONT=\"/bin/setfont\" \
@@ -159,6 +165,7 @@ endif
 endif
 endif
 endif
+endif
 
 rootbin_PROGRAMS = \
 	systemctl \
diff --git a/configure.ac b/configure.ac
index 62e8cdf..3ca9efa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -396,7 +396,7 @@ AM_CONDITIONAL(HAVE_XSLTPROC, test x"$XSLTPROC" != x)
 
 AC_PATH_PROG([M4], [m4])
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom, pbs or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -492,6 +492,11 @@ case $with_distro in
                 M4_DISTRO_FLAG=-DTARGET_MAGEIA=1
                 have_plymouth=yes
                 ;;
+        pbs)
+                SYSTEM_SYSVRCND_PATH=/etc
+                AC_DEFINE(TARGET_PBS, [], [Target is PBS])
+                M4_DEFINES=-DTARGET_PBS=1
+                ;;
         other)
                 ;;
         *)
@@ -550,6 +555,7 @@ AM_CONDITIONAL(TARGET_MANDRIVA, test x"$with_distro" = xmandriva)
 AM_CONDITIONAL(TARGET_MEEGO, test x"$with_distro" = xmeego)
 AM_CONDITIONAL(TARGET_ANGSTROM, test x"$with_distro" = xangstrom)
 AM_CONDITIONAL(TARGET_MAGEIA, test x"$with_distro" = xmageia)
+AM_CONDITIONAL(TARGET_PBS, test x"$with_distro" = xpbs)
 
 AM_CONDITIONAL(HAVE_PLYMOUTH, test "$have_plymouth" = "yes")
 AM_CONDITIONAL(HAVE_SYSV_COMPAT, test "$SYSTEM_SYSV_COMPAT" = "yes")
-- 
1.7.9.1

