From a5639b2b64bcf050617a60fd1793ba25bc6cc209 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Thu, 5 Jul 2012 10:06:00 +0200
Subject: [PATCH] Add PBS distribution support

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 Makefile.am  | 7 +++++++
 configure.ac | 8 +++++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 3b7ec0b..d1c5335 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -191,6 +191,12 @@ AM_CPPFLAGS += \
 	-DKBD_SETFONT=\"/bin/setfont\" \
 	-DDEFAULT_FONT=\"LatArCyrHeb-16\"
 else
+if TARGET_PBS
+AM_CPPFLAGS += \
+	-DKBD_LOADKEYS=\"/usr/bin/loadkeys\" \
+	-DKBD_SETFONT=\"/usr/bin/setfont\" \
+	-DDEFAULT_FONT=\"default8x16\"
+else
 AM_CPPFLAGS += \
 	-DKBD_LOADKEYS=\"/bin/loadkeys\" \
 	-DKBD_SETFONT=\"/bin/setfont\" \
@@ -202,6 +208,7 @@ endif
 endif
 endif
 endif
+endif
 
 # ------------------------------------------------------------------------------
 rootbin_PROGRAMS = \
diff --git a/configure.ac b/configure.ac
index dfa34c8..b2021fd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -500,7 +500,7 @@ AM_CONDITIONAL(ENABLE_MANPAGES, [test "$have_manpages" = "yes"])
 AC_PATH_PROG([XSLTPROC], [xsltproc])
 AM_CONDITIONAL(HAVE_XSLTPROC, test x"$XSLTPROC" != x)
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, ubuntu, arch, gentoo, slackware, altlinux, mandriva, meego, mageia, angstrom, pbs or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -589,6 +589,11 @@ case $with_distro in
                 AC_DEFINE(TARGET_MAGEIA, [], [Target is Mageia])
                 M4_DEFINES=-DTARGET_MAGEIA=1
                 ;;
+        pbs)
+                SYSTEM_SYSVRCND_PATH=/etc
+                AC_DEFINE(TARGET_PBS, [], [Target is PBS])
+                M4_DEFINES=-DTARGET_PBS=1
+                ;;
         other)
                 ;;
         *)
@@ -642,6 +647,7 @@ AM_CONDITIONAL(TARGET_MANDRIVA, test x"$with_distro" = xmandriva)
 AM_CONDITIONAL(TARGET_MEEGO, test x"$with_distro" = xmeego)
 AM_CONDITIONAL(TARGET_ANGSTROM, test x"$with_distro" = xangstrom)
 AM_CONDITIONAL(TARGET_MAGEIA, test x"$with_distro" = xmageia)
+AM_CONDITIONAL(TARGET_PBS, test x"$with_distro" = xpbs)
 
 AM_CONDITIONAL(HAVE_SYSV_COMPAT, test "$SYSTEM_SYSV_COMPAT" = "yes")
 
-- 
1.7.11.1

