diff --git a/configure b/configure
index 06e2f2f..79c6244 100755
--- a/configure
+++ b/configure
@@ -9487,7 +9487,7 @@ $as_echo "$LIBPAKCHOIS" >&6; }
 
 
 
-    if test "$ac_cv_pakchois" != yes; then
+    if test "$ac_cv_libpakchois" != yes; then
       included_pakchois=yes
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
   ***
diff --git a/lib/configure b/lib/configure
index 0b64077..7b40829 100755
--- a/lib/configure
+++ b/lib/configure
@@ -14589,7 +14589,7 @@ $as_echo "$LIBPAKCHOIS" >&6; }
 
 
 
-    if test "$ac_cv_pakchois" != yes; then
+    if test "$ac_cv_libpakchois" != yes; then
       included_pakchois=yes
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
   ***
diff --git a/libextra/configure b/libextra/configure
index 93c8ab3..8123fac 100755
--- a/libextra/configure
+++ b/libextra/configure
@@ -13877,7 +13877,7 @@ $as_echo "$LIBPAKCHOIS" >&6; }
 
 
 
-    if test "$ac_cv_pakchois" != yes; then
+    if test "$ac_cv_libpakchois" != yes; then
       included_pakchois=yes
       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
   ***
