diff --git a/build-aux/ltmain.sh b/build-aux/ltmain.sh
index aa5624c..0b11ae2 100755
--- a/build-aux/ltmain.sh
+++ b/build-aux/ltmain.sh
@@ -6907,7 +6907,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      add_dir="-L$lt_sysroot$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
diff --git a/lib/build-aux/ltmain.sh b/lib/build-aux/ltmain.sh
index aa5624c..0b11ae2 100755
--- a/lib/build-aux/ltmain.sh
+++ b/lib/build-aux/ltmain.sh
@@ -6907,7 +6907,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      add_dir="-L$lt_sysroot$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
diff --git a/libextra/build-aux/ltmain.sh b/libextra/build-aux/ltmain.sh
index aa5624c..0b11ae2 100755
--- a/libextra/build-aux/ltmain.sh
+++ b/libextra/build-aux/ltmain.sh
@@ -6907,7 +6907,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
+	      add_dir="-L$lt_sysroot$libdir"
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
