diff --git a/Makefile.am b/Makefile.am
index dd4bf17..2fcb80c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,7 +23,11 @@ DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --disable-valgrind-tests
 
 SUBDIRS = gl lib extra po
 
-SUBDIRS += src doc tests
+SUBDIRS += src tests
+
+if ENABLE_DOC
+SUBDIRS += doc
+endif
 
 if HAVE_GUILE
 SUBDIRS += guile
diff --git a/configure.ac b/configure.ac
index a54988e..daf9583 100644
--- a/configure.ac
+++ b/configure.ac
@@ -62,6 +62,12 @@ fi
 AM_CONDITIONAL(ENABLE_CXX, test "$use_cxx" != "no")
 AM_CONDITIONAL(WANT_TEST_SUITE, [test -f tests/suite/mini-eagain2.c])
 
+# documentation
+AC_ARG_ENABLE(doc,
+  AS_HELP_STRING([--disable-doc], [unconditionally disable documentation]),
+    [enable_doc=$enableval], [enable_doc=yes])
+AM_CONDITIONAL(ENABLE_DOC, test "$enable_doc" != "no")
+
 dnl Detect windows build
 case "$host" in
   *mingw32* | *mingw64*)
