diff --git a/Makefile.am b/Makefile.am
index 14c2642..12bb5d3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -21,12 +21,16 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc --disable-valgrind-tests
 
-SUBDIRS = gl lib extra po src doc tests
+SUBDIRS = gl lib extra po src tests
 
 if HAVE_GUILE
 SUBDIRS += guile
 endif
 
+if ENABLE_DOC
+SUBDIRS += doc
+endif
+
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4
 
 EXTRA_DIST = cfg.mk maint.mk .clcopying
diff --git a/configure.ac b/configure.ac
index 2d3a359..df3ec88 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,6 +68,12 @@ fi
 AM_CONDITIONAL(ENABLE_CXX, test "$use_cxx" != "no")
 AM_CONDITIONAL(WANT_TEST_SUITE, [test -f tests/suite/mini-eagain2.c])
 
+# documentation
+AC_ARG_ENABLE(doc,
+  AS_HELP_STRING([--disable-doc], [unconditonally disable documentation]),
+    [enable_doc=$enableval], [enable_doc=yes])
+AM_CONDITIONAL(ENABLE_DOC, test "$enable_doc" != "no")
+
 dnl Detect windows build
 case "$host" in
   *mingw32* | *mingw64*)
