From f5909df5cb68a9829c5793c0240b506da7863060 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Mon, 19 Mar 2012 12:26:24 +0100
Subject: [PATCH] Check for freetype using pkg-config

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 configure.in |   30 +++++++++++++++++-------------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/configure.in b/configure.in
index da6ef95..ba19dea 100644
--- a/configure.in
+++ b/configure.in
@@ -158,22 +158,26 @@ AC_DEFINE_UNQUOTED(USE_ICONV,$use_iconv,[Use iconv.])
 # Checks for FreeType
 #
 
-AC_ARG_WITH(freetype-config, [  --with-freetype-config=PROG   Use FreeType configuration program PROG], freetype_config=$withval, freetype_config=yes)
-
-if test "$freetype_config" = "yes"; then 
-	AC_PATH_PROG(ft_config,freetype-config,no)
-	if test "$ft_config" = "no"; then
-		AC_MSG_ERROR([You must have freetype installed; see http://www.freetype.org/])
+AC_ARG_WITH(freetype-config, [  --with-freetype-config=PROG   Use FreeType configuration program PROG], freetype_config=$withval, freetype_config=no)
+
+if test "$freetype_config" != "no"; then
+	if test "$freetype_config" = "yes"; then
+		AC_PATH_PROG(ft_config,freetype-config,no)
+		if test "$ft_config" = "no"; then
+			AC_MSG_ERROR([You must have freetype installed; see http://www.freetype.org/])
+		fi
+	else
+		ft_config="$freetype_config"
 	fi
-else
-	ft_config="$freetype_config"
-fi
 
-FREETYPE_CFLAGS="`$ft_config --cflags`"
-FREETYPE_LIBS="`$ft_config --libs`"
+	FREETYPE_CFLAGS="`$ft_config --cflags`"
+	FREETYPE_LIBS="`$ft_config --libs`"
 
-AC_SUBST(FREETYPE_LIBS)
-AC_SUBST(FREETYPE_CFLAGS)
+	AC_SUBST(FREETYPE_LIBS)
+	AC_SUBST(FREETYPE_CFLAGS)
+else
+	PKG_CHECK_MODULES(FREETYPE, freetype2)
+fi
 
 fontconfig_save_libs="$LIBS"
 fontconfig_save_cflags="$CFLAGS"
-- 
1.7.9.4

