diff --git a/drm_agpsupport.c b/drm_agpsupport.c
index a1af119..8b739d2 100644
--- a/drm_agpsupport.c
+++ b/drm_agpsupport.c
@@ -524,7 +524,7 @@ static int drm_agp_populate(struct drm_ttm_backend *backend,
 	DRM_DEBUG("Current page count is %ld\n", (long) mem->page_count);
 	mem->page_count = 0;
 	for (cur_page = pages; cur_page < last_page; ++cur_page)
-		mem->pages[mem->page_count++] = phys_to_gart(page_to_phys(*cur_page));
+		mem->pages[mem->page_count++] = page_to_phys(*cur_page);
 	agp_be->mem = mem;
 	return 0;
 }
diff --git a/drm_crtc.c b/drm_crtc.c
index f276439..fa00401 100644
--- a/drm_crtc.c
+++ b/drm_crtc.c
@@ -1811,6 +1811,8 @@ int drm_mode_attachmode_crtc(struct drm_device *dev, struct drm_crtc *crtc,
 		if (output->crtc == crtc)
 			drm_mode_attachmode(dev, output, mode);
 	}
+
+	return 0;
 }
 EXPORT_SYMBOL(drm_mode_attachmode_crtc);
 
diff --git a/drm_edid.c b/drm_edid.c
index 6307cdc..02d8bfa 100644
--- a/drm_edid.c
+++ b/drm_edid.c
@@ -39,7 +39,7 @@ int drm_get_acpi_edid(char *method, char *edid, ssize_t length)
 	if (obj && obj->type == ACPI_TYPE_BUFFER)
 		memcpy(edid, obj->buffer.pointer, obj->buffer.length);
 	else {
-		printk(KERN_ERR PREFIX "Invalid _DDC data\n");
+		printk(KERN_ERR "Invalid _DDC data\n");
 		status = -EFAULT;
 		kfree(obj);
 	}
