diff --git a/libkms/Makefile.am b/libkms/Makefile.am
index 0375ae4..2c75878 100644
--- a/libkms/Makefile.am
+++ b/libkms/Makefile.am
@@ -1,6 +1,7 @@
 AM_CFLAGS = \
 	$(WARN_CFLAGS) \
-	-I$(top_srcdir)/include/drm
+	-I$(top_srcdir)/include/drm \
+	-I$(top_srcdir)
 
 libkms_la_LTLIBRARIES = libkms.la
 libkms_ladir = $(libdir)
diff --git a/libkms/Makefile.in b/libkms/Makefile.in
index 3e0c473..efc206e 100644
--- a/libkms/Makefile.in
+++ b/libkms/Makefile.in
@@ -228,7 +228,8 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CFLAGS = \
 	$(WARN_CFLAGS) \
-	-I$(top_srcdir)/include/drm
+	-I$(top_srcdir)/include/drm \
+	-I$(top_srcdir)
 
 libkms_la_LTLIBRARIES = libkms.la
 libkms_ladir = $(libdir)
