Add pkg-config file.

From: Thierry Reding <thierry.reding@avionic-design.de>


diff --git a/Makefile.am b/Makefile.am
index 2acbb79..7075ee4 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -17,3 +17,5 @@ lib_LTLIBRARIES = libev.la
 libev_la_SOURCES = ev.c event.c
 libev_la_LDFLAGS = -version-info $(VERSION_INFO)
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = libev.pc
diff --git a/configure.ac b/configure.ac
index c6d1d7f..7828d30 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,5 +15,5 @@ fi
 
 m4_include([libev.m4])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile libev.pc])
 AC_OUTPUT
diff --git a/libev.pc.in b/libev.pc.in
new file mode 100644
index 0000000..8ef423f
--- /dev/null
+++ b/libev.pc.in
@@ -0,0 +1,9 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libev
+Description: a full-featured and high-performance event loop
+Version: @VERSION@
+Libs: -lev
