diff --git a/configure b/configure
index 0166829..6ce763a 100755
--- a/configure
+++ b/configure
@@ -2640,10 +2640,21 @@ if enabled libdc1394; then
     die "ERROR: No version of libdc1394 found "
 fi
 
-SDL_CONFIG="${cross_prefix}sdl-config"
-if "${SDL_CONFIG}" --version > /dev/null 2>&1; then
-    sdl_cflags=$("${SDL_CONFIG}" --cflags)
-    sdl_libs=$("${SDL_CONFIG}" --libs)
+PKG_CONFIG="${cross_prefix}pkg-config"
+if "${PKG_CONFIG}" --version sdl > /dev/null 2>&1; then
+    sdl_cflags=$("${PKG_CONFIG}" --cflags sdl)
+    sdl_libs=$("${PKG_CONFIG}" --libs sdl)
+    have_sdl=yes
+else
+    SDL_CONFIG="${cross_prefix}sdl-config"
+    if "${SDL_CONFIG}" --version > /dev/null 2>&1; then
+        sdl_cflags=$("${SDL_CONFIG}" --cflags)
+        sdl_libs=$("${SDL_CONFIG}" --libs)
+        have_sdl=yes
+    fi
+fi
+
+if test "x$have_sdl" = "xyes"; then
     check_func_headers SDL.h SDL_Init $sdl_cflags $sdl_libs &&
     check_cpp_condition SDL.h "(SDL_MAJOR_VERSION<<16 | SDL_MINOR_VERSION<<8 | SDL_PATCHLEVEL) >= 0x010201" $sdl_cflags &&
     enable sdl &&
