diff --git a/src/Makefile.am b/src/Makefile.am
index 598e77f..5d08cf2 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -51,25 +51,15 @@ endif
 
 AM_CFLAGS = \
 	-I$(top_srcdir)/src \
-	-I$(top_builddir)/src \
 	-I$(top_srcdir)/src/modules \
-	-I$(top_builddir)/src/modules \
 	-I$(top_srcdir)/src/modules/rtp \
-	-I$(top_builddir)/src/modules/rtp \
 	-I$(top_srcdir)/src/modules/gconf \
-	-I$(top_builddir)/src/modules/gconf \
 	-I$(top_srcdir)/src/modules/bluetooth \
-	-I$(top_builddir)/src/modules/bluetooth \
 	-I$(top_srcdir)/src/modules/oss \
-	-I$(top_builddir)/src/modules/oss \
 	-I$(top_srcdir)/src/modules/alsa \
-	-I$(top_builddir)/src/modules/alsa \
 	-I$(top_srcdir)/src/modules/raop \
-	-I$(top_builddir)/src/modules/raop \
 	-I$(top_srcdir)/src/modules/x11 \
-	-I$(top_builddir)/src/modules/x11 \
 	-I$(top_srcdir)/src/modules/jack \
-	-I$(top_builddir)/src/modules/jack \
 	$(PTHREAD_CFLAGS) -D_POSIX_PTHREAD_SEMANTICS \
 	$(LIBSAMPLERATE_CFLAGS) \
 	$(LIBSNDFILE_CFLAGS) \
diff --git a/src/Makefile.in b/src/Makefile.in
index 73a9959..9e675cb 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -2532,25 +2532,15 @@ PA_BINARY = $(bindir)/pulseaudio$(EXEEXT)
 ###################################
 AM_CFLAGS = \
 	-I$(top_srcdir)/src \
-	-I$(top_builddir)/src \
 	-I$(top_srcdir)/src/modules \
-	-I$(top_builddir)/src/modules \
 	-I$(top_srcdir)/src/modules/rtp \
-	-I$(top_builddir)/src/modules/rtp \
 	-I$(top_srcdir)/src/modules/gconf \
-	-I$(top_builddir)/src/modules/gconf \
 	-I$(top_srcdir)/src/modules/bluetooth \
-	-I$(top_builddir)/src/modules/bluetooth \
 	-I$(top_srcdir)/src/modules/oss \
-	-I$(top_builddir)/src/modules/oss \
 	-I$(top_srcdir)/src/modules/alsa \
-	-I$(top_builddir)/src/modules/alsa \
 	-I$(top_srcdir)/src/modules/raop \
-	-I$(top_builddir)/src/modules/raop \
 	-I$(top_srcdir)/src/modules/x11 \
-	-I$(top_builddir)/src/modules/x11 \
 	-I$(top_srcdir)/src/modules/jack \
-	-I$(top_builddir)/src/modules/jack \
 	$(PTHREAD_CFLAGS) -D_POSIX_PTHREAD_SEMANTICS \
 	$(LIBSAMPLERATE_CFLAGS) \
 	$(LIBSNDFILE_CFLAGS) \
