diff --git a/client/Makefile.am b/client/Makefile.am
index f01179a..b09f308 100644
--- a/client/Makefile.am
+++ b/client/Makefile.am
@@ -1,3 +1,5 @@
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/includes
+
 dist_sysconf_DATA = dhclient.conf
 sbin_PROGRAMS = dhclient
 dhclient_SOURCES = clparse.c dhclient.c dhc6.c \
@@ -11,8 +13,8 @@ EXTRA_DIST = $(man_MANS)
 
 dhclient.o: dhclient.c
 	$(COMPILE) -DCLIENT_PATH='"$(sbindir)"' \
-		   -DLOCALSTATEDIR='"$(localstatedir)"' -c dhclient.c
+		   -DLOCALSTATEDIR='"$(localstatedir)"' -c $(srcdir)/dhclient.c
 
 dhc6.o: dhc6.c
 	$(COMPILE) -DCLIENT_PATH='"$(sbindir)"' \
-		   -DLOCALSTATEDIR='"$(localstatedir)"' -c dhc6.c
+		   -DLOCALSTATEDIR='"$(localstatedir)"' -c $(srcdir)/dhc6.c
diff --git a/client/scripts/linux b/client/scripts/linux
index 49f0977..8b4d7a4 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # dhclient-script for Linux. Dan Halbert, March, 1997.
 # Updated for Linux 2.[12] by Brian J. Murrell, January 1999.
 # No guarantees about this. I'm a novice at the details of Linux
diff --git a/common/Makefile.am b/common/Makefile.am
index 8b2f2a9..003c49b 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I.. -DLOCALSTATEDIR='"@localstatedir@"'
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/includes -DLOCALSTATEDIR='"@localstatedir@"'
 
 noinst_LIBRARIES = libdhcp.a
 libdhcp_a_SOURCES = alloc.c bpf.c comapi.c conflex.c ctrace.c discover.c \
diff --git a/dhcpctl/Makefile.am b/dhcpctl/Makefile.am
index 35f1462..dc97eb2 100644
--- a/dhcpctl/Makefile.am
+++ b/dhcpctl/Makefile.am
@@ -1,3 +1,5 @@
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/includes
+
 bin_PROGRAMS = omshell
 lib_LIBRARIES = libdhcpctl.a
 noinst_PROGRAMS = cltest
diff --git a/dst/Makefile.am b/dst/Makefile.am
index 8937fe8..bd235c4 100644
--- a/dst/Makefile.am
+++ b/dst/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DMINIRES_LIB -DHMAC_MD5
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/includes -DMINIRES_LIB -DHMAC_MD5
 
 lib_LIBRARIES = libdst.a
 
diff --git a/minires/Makefile.am b/minires/Makefile.am
index 6fc1df1..85f0a59 100644
--- a/minires/Makefile.am
+++ b/minires/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DMINIRES_LIB -DHMAC_MD5
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/includes -DMINIRES_LIB -DHMAC_MD5
 
 noinst_LIBRARIES = libres.a
 
diff --git a/omapip/Makefile.am b/omapip/Makefile.am
index 6a3c39f..654a638 100644
--- a/omapip/Makefile.am
+++ b/omapip/Makefile.am
@@ -1,3 +1,5 @@
+INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/includes
+
 lib_LIBRARIES = libomapi.a
 noinst_PROGRAMS = svtest
 
diff --git a/relay/Makefile.am b/relay/Makefile.am
index ab95e84..4101df1 100644
--- a/relay/Makefile.am
+++ b/relay/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -DLOCALSTATEDIR='"@localstatedir@"'
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/includes -DLOCALSTATEDIR='"@localstatedir@"'
 
 sbin_PROGRAMS = dhcrelay
 dhcrelay_SOURCES = dhcrelay.c
diff --git a/server/Makefile.am b/server/Makefile.am
index efd3b04..095e511 100644
--- a/server/Makefile.am
+++ b/server/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I.. -DLOCALSTATEDIR='"@localstatedir@"'
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/includes -DLOCALSTATEDIR='"@localstatedir@"'
 
 dist_sysconf_DATA = dhcpd.conf
 sbin_PROGRAMS = dhcpd
