diff --git a/common/discover.c b/common/discover.c
index f2df447..27ebe90 100644
--- a/common/discover.c
+++ b/common/discover.c
@@ -684,8 +684,8 @@ next_iface6(struct iface_info *info, int *err, struct iface_conf_list *ifaces) {
 	for (i=0; i<16; i++) {
 		unsigned char byte;
                 static const char hex[] = "0123456789abcdef";
-                byte = ((index(hex, buf[i * 2]) - hex) << 4) |
-			(index(hex, buf[i * 2 + 1]) - hex);
+                byte = ((strchr(hex, buf[i * 2]) - hex) << 4) |
+			(strchr(hex, buf[i * 2 + 1]) - hex);
 		addr.sin6_addr.s6_addr[i] = byte;
 	}
 	memcpy(&info->addr, &addr, sizeof(addr));
