diff --git a/pkg.c b/pkg.c
index 7f252bd..0a60e52 100644
--- a/pkg.c
+++ b/pkg.c
@@ -483,6 +483,10 @@ string_list_to_string (GSList *list)
 	      g_string_append (str, pcsysrootdir);
 	      g_string_append (str, tmpstr+2);
 	    }
+	  else
+	    {
+	      g_string_append (str, tmpstr);
+	    }
 	}
       else 
 	{
@@ -1268,6 +1272,9 @@ packages_get_var (GSList     *pkgs,
       
       if (var)
         {
+          if ((pcsysrootdir != NULL) && (var[0] == G_DIR_SEPARATOR))
+            g_string_append (str, pcsysrootdir);
+
           g_string_append (str, var);
           g_string_append_c (str, ' ');                
           g_free (var);
