--- mtd-utils-1.5.0/Makefile	2012-05-07 09:19:39.000000000 +0200
+++ mtd-utils-1.5.0-dev/Makefile	2013-01-29 15:29:18.987935251 +0100
@@ -30,13 +30,17 @@
 	ubidetach ubinize ubiformat ubirename mtdinfo ubirsvol
 
 BINS = $(MTD_BINS)
-BINS += mkfs.ubifs/mkfs.ubifs
-BINS += $(addprefix ubi-utils/,$(UBI_BINS))
+ifneq ($(WITHOUT_UBI), 1)
+  BINS += mkfs.ubifs/mkfs.ubifs
+  BINS += $(addprefix ubi-utils/,$(UBI_BINS))
+endif
 SCRIPTS = flash_eraseall
 
 TARGETS = $(BINS)
 TARGETS += lib/libmtd.a
-TARGETS += ubi-utils/libubi.a
+ifneq ($(WITHOUT_UBI), 1)
+  TARGETS += ubi-utils/libubi.a
+endif
 
 OBJDEPS = $(BUILDDIR)/include/version.h
 
