From 23810f0b84c35239c9df67cacc98962acbbcd9d9 Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Fri, 15 Jun 2012 09:05:42 +0200
Subject: [PATCH] Fix out-of-tree build

Signed-off-by: Thierry Reding <thierry.reding@avionic-design.de>
---
 util/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/Makefile.am b/util/Makefile.am
index fa4474f..bd056b9 100644
--- a/util/Makefile.am
+++ b/util/Makefile.am
@@ -36,7 +36,7 @@ INIT_D_PATH = @INIT_D_PATH@
 
 install-exec-local:
 	$(MKDIR_P) $(DESTDIR)$(MOUNT_FUSE_PATH)
-	$(INSTALL_PROGRAM) $(srcdir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
+	$(INSTALL_PROGRAM) mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	$(MKDIR_P) $(DESTDIR)$(INIT_D_PATH)
 	$(INSTALL_SCRIPT) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
 	@if test -x /usr/sbin/update-rc.d; then \
-- 
1.7.10.4

