From df73a3638f733ca1b62b1e299ea29138a096f80f Mon Sep 17 00:00:00 2001
From: Thierry Reding <thierry.reding@avionic-design.de>
Date: Mon, 21 May 2012 14:10:17 +0200
Subject: [PATCH] Check for SDL via pkg-config

---
 configure.ac |   25 +------------------------
 1 file changed, 1 insertion(+), 24 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1b73966..ee6e2a4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -770,31 +770,8 @@ AM_CONDITIONAL(HAVE_LIBZ, test ! -z "$HAVE_ZLIB_H")
 AM_CONDITIONAL(HAVE_LIBJPEG, test ! -z "$HAVE_JPEGLIB_H")
 AM_CONDITIONAL(HAVE_LIBPNG, test ! -z "$HAVE_PNGLIB_H")
 
-
-SDLCONFIG="sdl-config"
-AC_ARG_WITH(sdl-config,
-[[  --with-sdl-config=FILE
-    Use the given path to sdl-config when determining
-    SDL configuration; defaults to "sdl-config"]],
-[
-    if test "$withval" != "yes" -a "$withval" != ""; then
-        SDLCONFIG=$withval
-    fi
-])
-
-if test -z "$with_sdl"; then
-	if $SDLCONFIG --version >/dev/null 2>&1; then
-		with_sdl=yes
-		SDL_CFLAGS=`$SDLCONFIG --cflags`
-		SDL_LIBS=`$SDLCONFIG --libs`
-	else
-		with_sdl=no
-	fi
-fi
+PKG_CHECK_MODULES(SDL, sdl, [with_sdl=yes], [with_sdl=no])
 AM_CONDITIONAL(HAVE_LIBSDL, test "x$with_sdl" = "xyes")
-AC_SUBST(SDL_CFLAGS)
-AC_SUBST(SDL_LIBS)
-
 
 # Check for GTK+. if present, build the GTK+ vnc viewer example
 PKG_CHECK_MODULES([GTK], [gtk+-2.0],,:)
-- 
1.7.10.2

