--- a/mount/umount.c	2007-03-06 13:26:41.000000000 +0100
+++ b/mount/umount.c	2007-03-06 13:26:48.000000000 +0100
@@ -31,7 +31,7 @@
 #include <arpa/inet.h>
 #endif
 
-#if defined(MNT_FORCE) && !defined(__sparc__) && !defined(__arm__)
+#if defined(MNT_FORCE) && !defined(__sparc__)
 /* Interesting ... it seems libc knows about MNT_FORCE and presumably
    about umount2 as well -- need not do anything */
 #else /* MNT_FORCE */
--- a/fdisk/llseek.c	2007-03-06 13:34:05.000000000 +0100
+++ b/fdisk/llseek.c	2007-03-06 13:34:16.000000000 +0100
@@ -16,7 +16,7 @@
 
 #ifdef HAVE_LLSEEK
 #include <syscall.h>
-
+#define my_llseek llseek
 #else	/* HAVE_LLSEEK */
 
 #if defined(__alpha__) || defined(__ia64__) || defined(__s390x__)
--- a/fdisk/sfdisk.c	2007-03-06 13:36:31.000000000 +0100
+++ b/fdisk/sfdisk.c	2007-03-06 13:36:44.000000000 +0100
@@ -172,7 +172,7 @@
  * Note: we use 512-byte sectors here, irrespective of the hardware ss.
  */
 #undef use_lseek
-#if defined (__alpha__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__)
+#if defined (__alpha__) || defined (__arm__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__)
 #define use_lseek
 #endif
 
