diff --git a/include/netlink/cache.h b/include/netlink/cache.h
index 415bb00..7101a3d 100644
--- a/include/netlink/cache.h
+++ b/include/netlink/cache.h
@@ -72,6 +72,8 @@ extern int			nl_cache_add(struct nl_cache *,
 extern int			nl_cache_parse_and_add(struct nl_cache *,
 						       struct nl_msg *);
 extern void			nl_cache_remove(struct nl_object *);
+extern int			nl_cache_request_full_dump(struct nl_sock *,
+							   struct nl_cache *);
 extern int			nl_cache_refill(struct nl_sock *,
 						struct nl_cache *);
 extern int			nl_cache_pickup(struct nl_sock *,
diff --git a/lib/cache.c b/lib/cache.c
index e99e9d2..8527f71 100644
--- a/lib/cache.c
+++ b/lib/cache.c
@@ -639,8 +639,7 @@ void nl_cache_set_flags(struct nl_cache *cache, unsigned int flags)
  *
  * @return 0 on success or a negative error code.
  */
-static int nl_cache_request_full_dump(struct nl_sock *sk,
-				      struct nl_cache *cache)
+int nl_cache_request_full_dump(struct nl_sock *sk, struct nl_cache *cache)
 {
 	if (cache->c_ops->co_request_update == NULL)
 		return -NLE_OPNOTSUPP;
-- 
1.8.3.1

