diff --git a/src/Makefile.in b/src/Makefile.in
index 9b08537..4e3fba4 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -59,6 +59,7 @@ PATHS='-DPPPOE_PATH="$(PPPOE_PATH)"' '-DPPPD_PATH="$(PPPD_PATH)"' \
 	'-DPPPOE_SERVER_OPTIONS="$(PPPOESERVER_PPPD_OPTIONS)"'
 
 CFLAGS= @CFLAGS@ $(LIC_INCDIR) $(DEFINES) $(LIC_DEFINE) $(PATHS) -Ilibevent
+LDFLAGS= @LDFLAGS@
 TARGETS=@TARGETS@
 PPPOE_SERVER_LIBS=$(LIC_LIBDIR) $(LIC_LIB)
 
@@ -67,16 +68,16 @@ all: $(TARGETS)
 	@echo "Type 'make install' as root to install the software."
 
 pppoe-sniff: pppoe-sniff.o if.o common.o debug.o
-	@CC@ -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o
+	@CC@ $(LDFLAGS) -o pppoe-sniff pppoe-sniff.o if.o common.o debug.o
 
 pppoe-server: pppoe-server.o if.o debug.o common.o md5.o libevent/libevent.a @PPPOE_SERVER_DEPS@
-	@CC@ -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent
+	@CC@ $(LDFLAGS) -o pppoe-server @RDYNAMIC@ pppoe-server.o if.o debug.o common.o md5.o $(PPPOE_SERVER_LIBS) -Llibevent -levent
 
 pppoe: pppoe.o if.o debug.o common.o ppp.o discovery.o
-	@CC@ -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o
+	@CC@ $(LDFLAGS) -o pppoe pppoe.o if.o debug.o common.o ppp.o discovery.o
 
 pppoe-relay: relay.o if.o debug.o common.o
-	@CC@ -o pppoe-relay relay.o if.o debug.o common.o
+	@CC@ $(LDFLAGS) -o pppoe-relay relay.o if.o debug.o common.o
 
 pppoe.o: pppoe.c pppoe.h
 	@CC@ $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o pppoe.o pppoe.c
diff --git a/src/configure b/src/configure
index a6f5d4b..8151ad7 100755
--- a/src/configure
+++ b/src/configure
@@ -6234,8 +6234,13 @@ echo "$as_me: WARNING: *** I will keep going, but it may not work." >&2;}
 	PPPD=pppd
 fi
 
+if test "${ac_cv_pppd_version+set}" = set; then
+	:
+else
+	ac_cv_pppd_version=`$PPPD --version 2>&1 | awk ' /version/ {print $NF}'`
+fi
 
-PPPD_VERSION=`$PPPD --version 2>&1 | awk ' /version/ {print $NF}'`
+PPPD_VERSION=$ac_cv_pppd_version
 
 case "$PPPD_VERSION" in
 1.*|2.0.*|2.1.*|2.2.*|2.3.0|2.3.1|2.3.2|2.3.3|2.3.4|2.3.5|2.3.6)
@@ -6272,7 +6277,11 @@ esac
 { echo "$as_me:$LINENO: checking packing order of bit fields" >&5
 echo $ECHO_N "checking packing order of bit fields... $ECHO_C" >&6; }
 if test "$cross_compiling" = yes; then
-  $ECHO "no defaults for cross-compiling"; exit 0
+  if test "${rpppoe_cv_pack_bitfields+set}" = set; then
+    $ECHO "(cached) $ECHO_C" >&6
+  else
+    $ECHO "no defaults for cross-compiling"; exit 0
+  fi
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
