diff --git a/Makefile.am b/Makefile.am
index 08bf5b8..835cf94 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,7 +1,7 @@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC=dbus-glib-1.pc
 
-SUBDIRS=dbus tools test doc
+SUBDIRS=dbus
 DIST_SUBDIRS=dbus tools test doc m4
 
 pkgconfigdir = $(libdir)/pkgconfig
diff --git a/Makefile.in b/Makefile.in
index 1a1029c..190fc56 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -311,7 +311,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC = dbus-glib-1.pc
-SUBDIRS = dbus tools test doc
+SUBDIRS = dbus
 DIST_SUBDIRS = dbus tools test doc m4
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index 02ef3a3..db91d6d 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 INCLUDES = \
 	-I$(top_srcdir)						\
diff --git a/dbus/Makefile.in b/dbus/Makefile.in
index f4fd885..f006170 100644
--- a/dbus/Makefile.in
+++ b/dbus/Makefile.in
@@ -368,7 +368,7 @@ target_vendor = @target_vendor@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . examples
+SUBDIRS = .
 INCLUDES = \
 	-I$(top_srcdir)						\
 	-I$(top_builddir)					\
