diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index 2650e8b..35f084b 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 INCLUDES = \
 	-I$(top_srcdir)						\
diff --git a/dbus/Makefile.in b/dbus/Makefile.in
index 7cae551..f7e9813 100644
--- a/dbus/Makefile.in
+++ b/dbus/Makefile.in
@@ -295,7 +295,7 @@ target_os = @target_os@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . examples
+SUBDIRS = .
 INCLUDES = \
 	-I$(top_srcdir)						\
 	-I$(top_builddir)					\
